\documentclass{article}
\usepackage[utf8]{inputenc}
\usepackage[english]{babel}
\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{setspace}
\usepackage{natbib} 
\usepackage{graphicx}
\usepackage{subfig}
\usepackage{comment}
\usepackage[backgroundcolor=pink,linecolor=red]{todonotes}
\usepackage{fullpage}
\usepackage[hidelinks]{hyperref}
\usepackage{xcolor}

\onehalfspacing

\newcommand{\E}{\mathbb{E}}
\renewcommand{\P}{\mathbb{P}}

\newcommand{\plr}[1]{\todo[linecolor=blue,backgroundcolor=blue!25,bordercolor=blue]{#1}}
\newcommand{\jgg}[1]{\todo[linecolor=green,backgroundcolor=green!25,bordercolor=black]{#1}}
\newcommand{\wac}[1]{\todo[linecolor=green,backgroundcolor=yellow!25,bordercolor=black]{#1}}

%%%%%%%%%%%
% for responses to reviewers
% set this to show line numbers and include responses to reviews or not
\newif\ifreviewresponses
\reviewresponsestrue  % include them
% \reviewresponsesfalse  % don't include them
\input{review-response-commands}
\newcommand{\responsefile}{review-responses.tex}  % name of the review reponses file


\begin{document}


\title{A few stickleback suffice for the transport of alleles to new lakes}
\author{Jared Galloway, William A. Cresko, and Peter Ralph}
\maketitle


\section*{Abstract}

Threespine stickleback populations provide a striking example of local adaptation to divergent habitats 
in populations that are connected by recurrent gene flow. 
These small fish occur in marine and freshwater habitats throughout the Northern Hemisphere, and in numerous cases the smaller freshwater populations have been established ``de novo'' from marine colonists. 
Independently evolved freshwater populations show similar phenotypes that have been shown to derive largely from the same standing genetic variants.
Geographic isolation prevents direct migration between the freshwater populations, strongly suggesting that these shared locally adaptive alleles are transported through the marine population.
However it is still largely unknown how gene flow, recombination, and selection jointly impact the standing variation that might fuel this adaptation.
Here we use individual-based, spatially explicit simulations to determine the levels of gene flow that best match observed patterns of allele sharing among habitats in stickleback.
We aim to better understand how gene flow and local adaptation in large metapopulations determine the speed of adaptation and re-use of standing genetic variation. 
In our simulations we find that repeated adaptation uses a shared set of alleles that are maintained at low frequency by migration-selection balance in oceanic populations. 
This process occurs over a realistic range of intermediate levels of gene flow that match previous empirical population genomic studies in stickleback. 
Examining these simulations more deeply reveals how lower levels of gene flow leads to slow, independent adaptation to different habitats, whereas higher levels gene flow leads to significant mutation load -- but an increased probability of successful population genomic scans for locally adapted alleles. 
Surprisingly, we find that the genealogical origins of most freshwater adapted alleles can be traced back to the original generation of marine individuals that colonized the lakes, as opposed to subsequent migrants. 
These simulations provide deeper context for existing studies of stickleback evolutionary genomics, and guidance for future empirical studies in this model.
More broadly, our results support existing theory of local adaptation but extend it by more completely documenting the genealogical history of adaptive alleles in a metapopulation.  


\section*{Introduction}

The canonical model for the genetics of adaptation has long been the sequential fixation of new mutations 
\citep{smith_haigh_1974, Endler_1977, Orr_2005}.
While it has proved to be a useful baseline for understanding the genetic variation we see in species today, this model is now rightfully understood as incomplete for many species in nature that have more complicated population structures 
\citep{Lai2152, Schrider2017}. 
In particular, empirical studies have increasingly identified the need to more deeply incorporate standing genetic variation into adaptation dynamics for metapopulations inhabiting an array of habitats 
\citep{Hermisson_2008, Barrett2008}. 
Populations experiencing diverse selective pressures while still exhibiting significant gene flow often result in more complex genomic signals that are still not fully understood 
\citep{Charlesworth_1997, Charlesworth_2003, Nosil_2009, Flaxman_2013, Samuk_2017}. 
Concurrently, a growing number of empirical studies have identified instances of convergent evolution using standing genetic variation 
\citep{Schrider2017, Barrett2008, Nelson_2018, Nelson_2019, Bassham_2018}. 
However, it is still not clear how variation in evolutionary processes - such as gene flow, recombination, selection and mutation -  can promote the maintenance and re-use of standing genetic variation, particularly during colonization and adaptation to new environments
\citep{Nelson_2018, Pritchard_2010, yeaman2011genetic, Schrider2017}. 
It is similarly unclear whether variation in these evolutionary processes can determine the genetic architecture of evolving traits via standing genetic variation. 
Theoretical work is mostly limited to one or two loci, or relies on approximations of uncertain validity
\citep[e.g.,][]{slatkin1975selection,petry1983effect,barton1986barrier,yeaman2011genetic,ralph2015standing}.
Therefore, detailed simulation models of specific systems can 
provide an important complement to empirical studies in the lab and field,
by helping us understand precisely how standing genetic variation might fuel local adaptation,
and what genomic signals we can expect to see.

The ancestral marine form of threespine stickleback fish (\textit{Gasterosteus aculeatus}) \revpoint{2}{1}
has given rise to hundreds of thousands if not millions of independently derived freshwater populations \revpoint{2}{3} \llabel{hundreds-thousands-or-more}
in recently de-glaciated regions around the Northern Hemisphere 
\citep{Bell_1994, Thompson_1997, Cresko_2007}. 
This model organism has provided some of the earliest data showing the heterogeneous nature of divergence across genomes \revpoint{2}{4}
and the much more extensive use of standing genetic variation than once thought 
\citep{schluter2009genetics, DeFaveri_2011, DeFaveri_2013, Roesti_2014, Nelson_2018, Bassham_2018, Nelson_2019, Hohenlohe_2010, Terekhanova_2014, Marques_2016}.  
While geographic isolation often prevents direct migration between freshwater populations, threespine stickleback in them \revpoint{2}{5}
frequently evolve similar phenotypes \citep{Cresko_2004, Colosimo_2004}. 
The most recent evolutionary genomic studies on stickleback document that while the overall dynamic of local adaptation to marine and freshwater habitats has been occurring for millions of years 
\citep{Nelson_2018}, independent local adaptation of marine individuals to freshwater environments has been observed to take place in just tens of generations \citep{Terekhanova_2014, Lescak2015evolution, Bassham_2018}. 
For example, in 1964 the Great Alaskan Earthquake caused an uplift of many islands and coastal regions throughout the Gulf of Alaska. 
Studies of stickleback populations on uplifted Middleton Island showed that newly created freshwater ponds were invaded by the surrounding marine population of stickleback which evolved the freshwater syndrome of phenotypes in less than 50 years 
\citep{Lescak2015evolution, Bassham_2018}. 
Amazingly, the portions of the genomes of these populations that showed increased divergence from the oceanic population
mirrored those previously found to differ between ocean and freshwater populations that
that have been geographically separated for thousands of years 
\citep{Hohenlohe_2010, Bassham_2018}. 


But how can evolution occur at such a rapid pace?
Waiting for new mutations to arise in each lake or pond would take much longer than the decades since the Alaskan earthquake
with any plausible target size.
Even more improbable is having divergence cluster in such similar genomic regions across independent populations. 
An alternative hypothesis
is that the majority of alleles important for freshwater adaptation are maintained in the marine individuals due to recurrent gene flow from freshwater back in to marine populations. 
\citet{schluter2009genetics} proposed a conceptual model they termed the ``transporter'' hypothesis to describe the process by which 
alleles beneficial in freshwater environments are maintained at migration-selection balance in the larger oceanic population and therefore available to be utilized during subsequent adaptation to new freshwater habitats.
(The alleles conferring adaptation to freshwater environments are thereby ``transported'' to and reassembled in new lakes.)
The first clear example of the global reuse of such alleles in stickleback was the gene \textit{eda} which has been shown to be an important regulator for the number of bony lateral plates 
\citep{Colosimo_2004, Cresko_2004, Colosimo_2005}. 
While the low lateral plate version of this gene arose millions of years ago, it is found in much younger freshwater ponds around the Northern Hemisphere 
\citep{O'Brian_2015}. 
More recently, genome-wide haplotype analyses have provided evidence that \textit{most} regions of the genome that distinguish marine-freshwater genetic differences share this pattern \citep{nelson2017ancient}. 

While the growing body of population genomic data on stickleback evolution supports the transporter hypothesis, a number of questions remain. 
What are the actual population sizes, migration rates, and fitness differentials consistent with this hypothesis? 
How many differentially selected alleles exist, how many are used at any one time, and how are they arranged within the genome? 
A curious natural history observation underlying many of these questions is the fact that some newly formed freshwater habitats, such as the ponds on Middleton Island, 
are quite small and presumably the number of initial marine migrants is few. The variation carried by these few initial migrants might therefore be a small subset of the total variation, and thus be insufficient to fuel adaptation without subsequent influx of alleles from additional generations of marine migrants carrying the remaining freshwater adaptive alleles.

Here, we use individual-based forward simulations implemented in SLiM that incorporates selection on a quantitative trait
explicitly determined additively from a realistically long genome to model the stickleback metapopulation and address these questions \citep{haller2017slim,haller2018slim3}. 
We ask how variation in amount of gene flow affects the genetic architecture of local adaptation to newly created freshwater ponds.
Because we record the entire genealogy of all alleles \citep{kelleher2018efficient}, 
we can determine the distribution and abundance of haplotypes across all marine and freshwater populations to know the timing and proportion of potentially adaptive alleles that are actually utilized in each population. 
In addition, we can document how these adaptive alleles are distributed across the genomes, and as a consequence determine how this may affect the efficacy of genome scans for between-habitat differentiation. 


\section*{Methods}

To explore these questions, we used SLiM \citep{haller2017slim,haller2018slim3} to implement forward-time simulation of populations of individuals with explicitly represented genomes in which selection acted upon a single continuous quantitative trait. 
The details and equations of the model were motivated by current understanding of threespine stickleback evolutionary, history and demography, as well as population genetics theory \citep{Gillespie_1998}. \llabel{clarify_equations}
The following model included divergent selection in the two habitats which had substantial spatial structure.
Certain aspects of the model remain simplistic due to computational constraints;
in particular, total population sizes are much smaller than in reality,
although we may capture a good picture of local dynamics. 

\paragraph{Habitat and geography}
Our simulations include two habitat types -- marine and freshwater -- defined by the nature of their selective pressures, each with 5,000 diploid individuals. 
The arrangement of these habitats, depicted in Figure~\ref{fig:Geo}, roughly models a set of freshwater habitats along a stretch of coastline. 
The marine habitat is a continuous, one-dimensional range of length 25 units, while the freshwater habitat is divided into 25 discrete subpopulations (which we call ``lakes''), each connected to the marine habitat at regularly spaced intervals (positions $i - 1/2$ for $1 \le i \le 25$).

Divergent selection is mediated by a single quantitative trait with different optima in marine and freshwater habitats. 
This situation roughly models the cumulative effect of the various phenotypes thought to be under divergent selection between the habitats, such as armor morphology, body size, craniofacial variation and opercle shape.
The optimal trait values in the marine and freshwater habitats are $+10$ and $-10$ respectively, and fitness of a fish with trait value $x_\text{ind}$ in a habitat with optimal value $x_\text{opt}$ is determined by a Gaussian kernel with standard deviation 15, i.e.,

\begin{align*}
    f(x_\text{ind}; x_\text{opt})
    &=
    \exp\left\{
        \frac{1}{2}
            \left(
            \frac{x_\text{ind}-x_\text{opt}}{15}
            \right)^2
        \right\} .
\end{align*}

Thus, the difference between an individual's trait value and the optimum determines that individual's fitness. 
We chose the difference between optima and strength of stabilizing selection in each habitat so that (a) around 10 (diploid, homozygous) mutations were sufficient to move from one optimum to the other, and (b) well-adapted fish from one habitat would have low, but nonzero, fitness in the other habitat.

\begin{figure}
    \begin{center}
          \includegraphics[width=0.8\linewidth]{Final_Plots/GeographyFigure.pdf}
          \caption{
            \textbf{Diagram of simulated populations:}
            a single, continuous, one-dimensional marine habitat (blue)
            is coupled to ``lakes'', in which stickleback randomly mate, at discrete locations with arrows representing migration patterns.
            After an initial period of 100K generations with 25 lakes,
            an additional 25 lakes are added (at the same set of locations) and populated with marine individuals
            to simulate the appearance of newly accessible freshwater habitats colonized by marine stickleback.
            The marine habitat, and each set of 25 lakes, each contain 5,000 individuals at all times.
            }
          \label{fig:Geo}
    \end{center}
\end{figure}

\paragraph{Genetic architecture of the trait}

Each individual carries two linear chromosomes, each of size $10^8$ loci,
with recombination events occurring at a intralocus rate of $10^{-8}$ every generation. \revpoint{2}{9}
%Mutations that can affect the trait under selection can occur at rate $10^{-10}$ per locus per generation \revpoint{2}{8}
%in ten regions of $10^5$ loci each, spread evenly along the chromosome. 
Mutations that can affect the trait under selection can occur at rate $10^{-8}$ with a target size of $10$ Kb, (spread evenly among $1$ Mb) across each individual genome. \revpoint{2}{8}
With 5000 (diploid) individuals per population, this equates to $\approx 1$ mutation, per population, per generation, which impact individual trait value. \llabel{mu_exp}
Each mutation in these regions is either additive, completely recessive, or completely dominant (with equal probability). 
Effect sizes for these mutations are chosen randomly from an exponential distribution with mean $1/2$, either positive or negative with equal probability. 
Individual trait values ($x_\text{ind}$) are determined additively from the diploid genotypes. 
Concretely, an individual that is heterozygous and homozygous for mutations at sets of loci $H$ and $D$ respectively has trait value $x_\text{ind} = \sum_{i \in H} h_i s_i + \sum_{j \in D} s_j$, where $h_i$ and $s_i$ are the dominance coefficient and the effect size of the mutation at locus $i$. Subsequent mutations at the same locus replace the previous allele. 

\paragraph{Population dynamics}

We use SLiM to simulate a Wright--Fisher population with non-overlapping generations \revpoint{2}{10}
and a fixed population size of 5,000 diploid individuals in each habitat. 
Each generation, the two parents of each new offspring are chosen proportional to their fitness 
(unlike actual stickleback, all individuals are hermaphroditic), and the contributing genomes are produced by Poisson recombination with an average of one crossover per chromosome per generation ($10^{-8}$ per locus per generation). 
Since the total population across \emph{all} 25 lakes is fixed at 5000, and the Wright--Fisher model assumes unrealistic global population regulation, we normalize the fitnesses of each individual so that approximately 200 offspring are generated in each lake, each generation. 
(A more realistic implementation would use SLiM's ``non-Wright-Fisher'' model type, \revpoint{2}{11}
which was not available when we wrote the simulations; but we do not expect the difference to affect results.) \llabel{realistic}
To do this, we divide fitness values of each freshwater individual by the mean fitness in their lake, so that the mean fitnesses of all lakes are equal before selection happens.

As depicted in Figure~\ref{fig:Geo} the marine population is continuous (unidimensional) while lakes are discrete. \llabel{continuous}
Dispersal occurs both locally along the coastline within the marine habitat, as well as between the marine habitat and the lakes,
with a lake--ocean migration rate denoted $m$. 
No migration occurs directly between the freshwater populations. 
All individual dispersal events can be thought of as occurring at the juvenile stage in the life cycle of the simulation. 
Each new individual in each habitat has parents from the other habitat with probability $m$ (in which case we call it a ``migrant''), and parents from the same habitat with probability $1-m$.
The first parent of each non-migrant individual in the freshwater habitat is chosen from the freshwater habitat proportional to fitness, and a mate is chosen from the same lake as the first, also proportional to fitness. The offspring then lives in the same lake as the parents. Parents for each non-migrant marine individual are chosen similarly:
first, a single parent is chosen proportionally to fitness in the marine habitat, and then a mate is chosen, also proportionally to fitness but re-weighted by a Gaussian function of the distance separating the two, with standard deviation $1/2$. Concretely, if the first parent is marine individual $i$, then marine individual $j$ is chosen as the mate with probability proportional to $f(x_j) \exp(-2d_{ij}^2)$, where $f(x_j)$ is the fitness of individual $j$ and $d_{ij}$ is the distance between the two locations. 
Finally, each new marine offspring is given a position displaced from the first parent's position by a random Gaussian distance with mean 0 and standard deviation 0.5, and reflected to stay within the habitat range.
Parents for each freshwater migrant are chosen in the same way as for non-migrant marine individuals, and are assigned to the lake nearest to the position of the first marine parent. Similarly, parents for each marine migrant are both chosen from the same lake as before, and the offspring is given a spatial location in the marine habitat at the location of the parent's lake. 

\paragraph{Colonization of newly formed lakes} 

To study how marine-derived populations adapt after colonizing newly appearing freshwater habitats, we introduce a new set of 25 lakes after 100,000 generations. 
These new lakes are populated with marine individuals to emulate a freshwater lake being colonized by oceanic stickleback that had previously been exchanging alleles with older freshwater populations. 
This creates two \emph{sets} of lakes, which along with the marine population have a total of 15,000 individuals. Since this introduction of lakes doubles the number of lake-to-marine immigrants, the probability that a new marine individual has freshwater parents is $2m$ instead of $m$.

\paragraph{Recording genealogical history}

We used SLiM's ability to record \emph{tree sequences} \citep{haller2018treesequence,Kelleher2016} to output the genealogical history of all individuals at the time of introduction of new lakes, at the time of adaptation, and at the end of the simulation. This allowed us to directly query the true origins of adaptive alleles. 
In addition, it allowed for much larger simulations by avoiding the computationally expensive task of simulating neutral mutations which were retroactively added to the gene trees at a rate of $10^{-8}$ per locus per generation, as described in \citet{kelleher2018efficient}.
The tree sequence output by each simulation allows us to explore the origin of the genetic basis of adaptation in the new lakes. 
To do this, we constructed the genealogical tree relating all extant chromosomes at each locus along the genome. 
Using these trees we classified each adaptive allele, in each genome in the new lakes at the time of adaptation, into four categories:

\begin{enumerate}
    \item \emph{a ``De novo" allele:}
        deriving from a new mutation that occurred in a new lake.
    \item \emph{a ``Migrant" allele:} 
        deriving from a migrant not in the initial generation that colonized the lake
    \item \emph{a ``Captured" allele:}
        present in initial colonists of the new lake, 
        and both common (above 50\%) in the original lakes,
        and uncommon (below 50\%) in the ocean.
    \item \emph{a ``Marine" allele:} 
        present in initial colonists of the new lake, 
        but not a ``captured'' allele.
\end{enumerate}

The proportion of trait-affecting alleles in new lakes that fall in these categories measures the degree to which selection in the new environments made use of (1) new mutation, (2) post-colonization migration, (3) standing variation at migration--selection balance, and (4) standing variation at mutation--selection balance.

We used neutral mutations to calculate $F_{ST}$ for each locus, which we then averaged in windows as a measure of between-population relative differentiation. 
Concretely, if $p_f$ and $p_m$ are the frequencies of a given mutant allele in the freshwater and marine habitats, respectively, and $\bar p = (p_f + p_m)/2$, 
then we compute $F_{ST}$ for that mutation as $1 - (p_f (1-p_f) + p_m (1-p_m))/ (2 \bar p (1-\bar p))$ \citep{Gillespie_1998}. \llabel{Fst}

\subsection*{Data availability}

The source code used to run and analyze the simulations in this paper
are available at \url{https://github.com/jgallowa07/SticklebackPaper}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Results \& Discussion}

To observe the impact of gene flow on selection in the new lakes, we varied the ocean--lake migration rate, $m$, across separate simulations from $5 \times 10^{-5}$ to $5 \times 10^{-1}$. 
Below, we often refer to these migration rates in terms of the number of migrants per lake, per generation, which we denote $M$. Since each lake contains 200 individuals, $M=200m$. 
Many aspects of adaptation changed substantially across this range, including the speed of adaptation, degree of sharing of adaptive alleles between lakes, and the population genetic signals left behind. 
At very low rates of gene flow, each new lake's population adapted almost completely independently through \emph{de novo} mutation, which took a very long time ($\approx 20,000$ generations). 
At very high rates of gene flow, local adaptation was constrained by the large influx of locally maladaptive alleles. % \citep{bolnick2007natural}. 
Between these two extremes, genetic variation that allowed adaptation to freshwater habitats could move relatively easily between lakes. 
Perhaps surprisingly, only a few migrants per generation from the lakes to the ocean were needed to maintain sufficient genetic variation in the ocean to dramatically accelerate adaptation in new lakes.

\subsection*{Rapid local adaptation at intermediate gene flow rates}

\begin{figure}
    \begin{center}
          \includegraphics{Final_Plots/Pheno_Dist.pdf}
          \caption{
            Distribution of mean individual trait values across generations of the simulation,
            for different migration rates.
            The dashed pink and purple lines at $\pm 10$ give the optimum phenotypes
            in the marine and freshwater environments, respectively.
        }
          \label{fig:MeanPhenotype}
    \end{center}
\end{figure}

\begin{figure}
    \begin{center}
        \includegraphics{Final_Plots/Pheno_Time.pdf}
          \caption{ 
        Mean individual trait values in the marine habitat (red line),
                the original lakes (tan lines; average thicker),
                and the new lakes (black lines; average thicker),
                across the course of two simulations, with migration rates of
                \textbf{(A)} $M=0.01$ and
                \textbf{(B)} $M=0.1$ 
                migrants per lake per generation, respectively.
                Optimal trait values in the two habitats are at $\pm 10$.
                    Analogous plots for other migration rates
                    are shown in Figure \ref{fig:phenotype_time_supp}.
        }
          \label{fig:phenotype_time}
    \end{center}
\end{figure}

Local adaptation occurred in all simulations with the exception of the highest gene flow at which half of each population was composed of migrants, 
as can be seen by the mean trait values of Figure~\ref{fig:MeanPhenotype}.
Figures~\ref{fig:phenotype_time} and \ref{fig:phenotype_time_supp} show how mean trait values in freshwater and marine populations diverged over time, until the trait means were close to the optimal values in each habitat. 
Establishment of new alleles in the lakes during the first few thousand generations manifests as jumps in the mean trait value which move the trait by an amount of order 1 every few hundred generations (Fig. \ref{fig:phenotype_time}).
At the lowest rate of gene flow, $M = 0.01$, differences at around 20 commonly polymorphic sites (about 10 that shift the trait in each direction) were responsible for most of the adaptive differences between freshwater and marine habitats. 
As expected, increasing migration rate decreased differentiation between habitats: as seen in Figure~\ref{fig:Fst}, $F_{ST}$ between marine and freshwater habitats at neutral sites steadily declines as migration increases.  \revpoint{1}{1}

\begin{figure}
    \begin{center}
        \includegraphics[width=\textwidth]{Final_Plots/Fst_Genome_faa.pdf}
          \caption{
        Average $F_{ST}$ in windows of 500Mb between:
        \textbf{(top)} marine habitat and old lakes;
        \textbf{(middle)} marine habitat and new lakes; and
        \textbf{(bottom)} old and new lakes.
        Each plot shows $F_{ST}$ values for a separate simulation,
        with columns corresponding to increasing gene flow from left to right.
        $F_{ST}$ is calculated between all marine individuals, and all lakes pooled together.
        All locations of pre-existing freshwater adapted alleles have been highlighted by 
        partially transparent pink dashed vertical lines, so darker shades of pink show more 
        freshwater adapted alleles at that location.
     } \label{fig:Fst}
    \end{center}
\end{figure}

Adaptation occurred much more quickly at higher migration rates, both in the old and new sets of lakes. We measured this ``time to adaptation'' as the number of generations until average trait values in old and new lakes were within 0.5 of each other, shown in Figure~\ref{fig:TimeToAdaptation} for different rates of gene flow. Adaptation of new lakes took over 18,000 generations at the lowest rate of $M = 0.01$, while at $M = 1$ migrant per lake per generation, new lakes adapted in just under 60 generations. 

\begin{figure}
    \begin{center}
          \includegraphics{Final_Plots/Time_Adapt.pdf}
          \caption{
            Time to adaptation as a function of migration rate.
            The time to adaptation is measured as the number of generations until
            the introduced population's mean phenotype 
            comes within 0.5 of the original lakes average phenotype. 
            Each point represents a single simulation run.
            (Adaptation did not occur at the highest rate of gene flow.)
        } \label{fig:TimeToAdaptation}
    \end{center}
\end{figure}

\subsection*{Widespread allele sharing between lakes}

Adaptive alleles were shared between lakes at many rates of gene flow, but not at the lowest.
At low migration rates, the \emph{initial} period of adaptation takes roughly 25 times longer for lakes than it does for the ocean. 
This difference occurs because in the absence of gene flow, each lake must wait for its own novel mutations to arise in order to adapt. 
Because the marine habitat is continuous, with 25 times more individuals than any one lake, there is a much larger influx of new mutations to be selected upon. 
At higher migration rates, greater mixing allows the initial lakes to share alleles instead of developing their own genetic basis for adaptation. 

To investigate in more depth how locally adaptive alleles found in the original lakes are shared among newly derived lakes, as well as how they spread to the new lakes, we defined and tracked the distribution of ``pre-existing freshwater adapted alleles'' at the beginning of each generation. 
To be considered in this category, an allele must participate in the genetic basis of local adaptation for at least one of the original lakes. 
Concretely, these are any non-neutral mutations whose frequency is above 50\% in at least one original lake and below 50\% in the marine habitat. 
(So, ``captured'' alleles as defined above are pre-existing, but not all pre-existing alleles are captured.)
Figure \ref{fig:MPFAI}A shows the distribution of the number of these alleles across generations. At $M = 0.01$, each lake has a private set of about 10 mutations nearly fixed in that lake but not elsewhere: new lakes independently acquire new adaptive alleles rather than pre-existing ones. 
At $M = 0.1$, we again observe the original lakes adapting nearly independently from each other, but now the new lakes adapt using pre-existing alleles present in the original set of lakes. 
Concurrently, the average marine individual carries $\approx 2$ pre-existing freshwater adapted alleles, standing variation which was nearly absent at $M=0.01$. As migration rate increases past this, the total number of pre-existing freshwater adapted alleles declines (Figure \ref{fig:MPFAI}C). Interestingly, the frequency of these alleles in the ocean
stays relatively constant across the reasonable rates of gene flow.

Figure~\ref{fig:MPFAI}B shows the distribution, through time, of the mean \emph{percentage} of currently-defined freshwater adapted alleles that each genome in each of the populations carries. 
If all individuals across lakes carried the same set of alleles determining their trait value, this would be 100\%. At the lowest migration rate ($M = 0.01$), each genome in the original lakes have almost exactly~$1/25^{th}$ of the total number of pre-existing freshwater adapted alleles -- this is because each one of the 25 lakes has adapted with a unique set of alleles. 
Since these are \emph{pre-existing} alleles, the value is zero for introduced lakes. 
Figure \ref{fig:MPFAI}A shows us that at $0.1$ migrants per lake per generation and above, the average individual across the new lakes has nearly the same amount of pre-existing freshwater adapted alleles as individuals across the old lakes. 
As expected, the genetic basis of the freshwater phenotype seems to simplify as migration increases -- higher rates of migration allow adaptive alleles of larger effect to travel more efficiently through the population, even though they are deleterious in the ocean. \revpoint{1}{1}

The numbers in Figure~\ref{fig:MPFAI} strongly suggest that the dramatic increase in speed of local adaptation we observed above occurs because higher gene flow allows sharing of freshwater alleles among populations.
We confirmed this by using recorded tree sequences to identify the origin of each trait-affecting allele in each individual in the new lakes, as defined in the Methods. 
Figure~\ref{fig:Origin} shows that at the lowest rate of gene flow the majority of adaptive alleles are derived from \emph{``de novo''} mutation. 
As gene flow increases, a larger fraction of adaptive alleles derive from pre-existing variation in the marine population at the time of introduction. 
In other words, greater mixing at higher migration rates allows lakes to share alleles instead of developing their own genetic basis of adaptation.

\begin{figure}
    \begin{center}
          \includegraphics{Final_Plots/Freshwater_Alleles.pdf}
          \caption{
            Amount of standing freshwater variation by habitat, across migration rates.
            Each plot counts ``pre-existing freshwater adapted alleles'',
            that are common in the original lakes but rare in the ocean
            (see text for definition).
            \textbf{(A)} Mean number of these alleles per individual.
            \textbf{(B)} Mean percentage of these alleles per individual.
            \textbf{(C)} Total number of these alleles (so, $B = A/C$).
            The number of alleles meeting these conditions changes over the course of the simulation,
            and each plot shows distributions of these values throughout each simulations.
            The horizontal axis shows $M$, the mean number of migrants per lake per generation.}
        \label{fig:MPFAI}
    \end{center}
\end{figure}


\begin{figure}
    \begin{center}
    \includegraphics{Final_Plots/Allele_Origin_2.pdf}
    \caption{ 
        \textbf{(Origin of adaptive alleles:)}
        Each bar plot shows the origins
        of all trait-affecting alleles above frequency 50\% in at least one new lake,
        classified as
        \textbf{(red)} new mutations,
        \textbf{(blue)} post-colonization migrants,
        \textbf{(green)} ``captured'' from pre-existing lakes, or
        \textbf{(orange)} standing marine variation.
        See Methods for precise definitions of these categories.
    } \label{fig:Origin}
    \end{center}
\end{figure}

While increased migration allows sharing of adaptive alleles among lakes, at $M = 10.0$ migrants per lake per generation
the constant influx of alleles between the habitats creates substantial migration load. 
This level of gene flow only replaces 5\% of each population each generation with migrants from the other habitat, 
but is sufficient to shift the mean trait values to nearly half their optimal values, as seen in Figure \ref{fig:MeanPhenotype}. 


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{The efficacy of genomic scans of selection depends on gene flow}

\revpoint{2}{13}
Here, we take a closer look at the genomic architecture of local adaptation between the two habitats. 
Can measures of local differentiation such as $F_{ST}$ be used to identify the causal loci? Figure \ref{fig:Fst} shows plots along the genome of average per-locus $F_{ST}$ values in 500bp windows between the marine habitat and all freshwater habitats pooled together. 
Higher rates of dispersal showed more distinct $F_{ST}$ peaks over polymorphic loci, while ``background'' levels of $F_{ST}$ increase as gene flow decreases, swamping out this signal until the regions under selection are indistinguishable. This is likely the contribution of two separate forces of natural selection: 
first, stronger genetic drift with less migration leads to higher background $F_{ST}$, and second, greater sharing of adaptive alleles providing a shared signal across populations.

At first glance, this suggests that genome scans for local adaptation based purely on measures of differentiation will only be successful given enough migration between habitats. 
But how many of these peaks are actually underlying trait differences that form the basis for local adaptation? 
To quantify this, Figure \ref{fig:Power_FP} shows the power and true positive rates that would be obtained by an $F_{ST}$ cutoff that declared everything above a certain value to be a causal locus. 

What we observe most plainly in this graph is that $M = 1.0$ and $M = 10.0$ migrants per lake per generation 
are the only rates of dispersal at which a large percentage of peaks (above $F_{ST} = 0.05$) 
actually lie on top of regions which impact trait differences.
Unfortunately, the low statistical power for all rates of $M$ would seem to suggest that many regions which impact trait differences do not appear as $F_{ST}$ peaks. 
In other words, with high rates of dispersal between population, $F_{ST}$ peaks may reliably identify causal loci, 
but not all causal loci will appear as $F_{ST}$ peaks.
It's important to note that $F_{ST}$ in Figures~\ref{fig:Fst} and \ref{fig:Power_FP} was calculated after pooling together all lakes and all marine individuals.
For comparison, Figures~\ref{fig:fst_5lakes} \& \ref{fig:Power_FP_5lakes} shows $F_{ST}$ calculated with individuals only in the first fifth of the range (5 lakes and corresponding marine habitat).
These show a substantially lower true positive rate, as population-specific noise swamps any signal between the lakes.

As has been suggested elsewhere, introgression aids identification of locally adaptive loci from scans for differentiation.
% \plr{CITE}
This is generally because shared adaptive alleles provide something akin to independent replicates,
allowing signal to emerge above the noise created by genetic drift.
However, the required sample sizes and conditions on levels of admixture may be restrictive in practice.
More generally, the difficulty of correctly identifying causal loci in even our most ideal situation
calls for caution (and proof-of-concept simulations) in designing and carrying out scans for alleles underlying locally adaptive, polygenic traits.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Alleles for freshwater adaptation are mostly present in the initial generation}

We have thus far found that the speed of adaptation depends strongly on the degree to which alleles can be shared between populations. 
However, the \textit{origin} of the alleles underlying the phenotype is still unknown. 
What is the nature of selection on standing genetic variants? 
In our simulations migration from marine individuals into the new lakes continues throughout introduction and adaptation: but is consistent influx of standing genetic variation a necessity for rapid adaptation of the population?
To answer this, we traced the genealogy of \textit{all} adaptive alleles in \textit{all} individuals from the introduced lakes, after local adaptation.
Surprisingly, for all cases with the exception of $M = 0.01$, we found that the majority of adaptive allele origins traced back the original generation of inhabitants in the lake (Figure \ref{fig:Origin}). \revpoint{1}{3}
In our simulations,
any reasonable subset of the ancestral population has the potential for rapid adaptation without the need post-colonization hybridization events.

If effectively capturing standing genetic variants is the key to rapid adaptation, as we have presented thus far, why is there a large difference in speed of adaptation between dispersal rates of $M = 0.1$ and $M = 1.0$ seen in Figure \ref{fig:TimeToAdaptation}?
%Figure~\ref{fig:Origin} shows both migration levels being able to effectively capture and propagate the standing genetic variants, and
%Figure~\ref{fig:MPFAI}A shows that with $M = 0.1$ migrants per lake per generation, there is substantial sharing of alleles between populations.
%So why does it take $M = 0.1$ scenario nearly 2,000 generations to adapt. 
This is surprising because we see a similar quantity of allele sharing at  $M = 1.0$, but much more rapid adaptation ($\approx 60$ versus 2000 generations).
%A reasonable explanation for this 33-fold discrepancy is that new lakes must wait for an additional influx of freshwater alleles through migration beyond what is present in the initial generation. However, Figure~\ref{fig:Origin} shows this is not the case: at both $M = 0.1$ and $M = 1.0$, the vast majority of alleles responsible for local adaptation came from individuals in the original generation (``captured'').

\begin{table}[ht]
    \centering
    \begin{tabular}{rrrrr}
      \hline
     & M=0.01 & M=0.1 & M=1 & M=10 \\ 
      \hline
          best & 2.77 & -12.02 & -21.02 & -7.02 \\ 
      % expected & 4.22 & -4.78 & -14.64 & -5.47 \\ 
       \hline
    \end{tabular}
    \caption{
        Haplotypic variation present in the new lakes at time of colonization,
        across rates of gene flow,
        \textbf{``Best''} 
        quantified as the most negative trait value achievable
        with intact haplotypes, averaged across populations
        (see text for details).
    } \label{tab:linkage}
\end{table}

A possible explanation is that at lower rates of gene flow, freshwater alleles present as standing variation in the marine habitat are more tightly linked to marine alleles. 
This implies that adaptation in the new lakes must wait for recombination to separate freshwater and marine adapted alleles. 
This might be expected at low dispersal rates since freshwater alleles may need to be ``masked'' by nearby compensatory alleles to remain in the ocean for long periods of time. 
Reversing this ``masking'' then slows the process of adaptation that uses this genetic variation.

%\plr{cite \citet{yeaman2011genetic}?}

In other words, higher dispersal from freshwater to the ocean maintains relatively intact freshwater haplotypes that can be more easily rebuilt in the marine environments. Recall that trait-affecting mutations only occur in relatively small regions of $10^5$ loci in which recombination occurs only once in every one thousand meioses. This implies that even if there is a sufficient amount of variation in the initial population of a lake to shift the trait from the marine optimum ($+10$) to the freshwater optimum ($-10$), rebuilding the most beneficial haplotype may prove to be a non-trivial task for selection.
%these may be tightly linked to alleles that counteract their effect.
For example, suppose there are 10 variants segregating at low frequency with effect size $-1$ each, but each is paired with a compensatory allele with effect size $+1$. Each local haplotype is therefore neutral. This might also explain why at $M=0.1$, marine individuals still hold on average several alleles that shift the trait in the freshwater direction
(Figure \ref{fig:MPFAI}A).
%(and so likely to be at higher frequency in the marine population).

%Indeed, this appears to be the case.
To quantify the genetic variation available \emph{without} recombination within the ten genomic regions, we first found, within each population, the haplotype with the largest net negative effect at each of the ten genomic regions. Summing these ten numbers, we get the maximum amount that selection could move the population in the freshwater direction without recombining {within} these regions. The mean of this value across the 25 lake populations is shown in Table~\ref{tab:linkage} -- typical populations at $M=0.1$ migrants per lake per generation could shift to a phenotype of -12.02
(and so have sufficient variation to adapt without recombination), but the ``best'' haplotypes in the populations at $M=1.0$ have effect sizes nearly twice as big at a mean total of -21.02.
The amount of variation available at $M=10.0$ is lower (only -7.02), presumably because of migration load, while at $M=0.01$ almost no alleles with negative effect are present.
Note that some of these haplotypes will likely be lost to drift -- indeed, if they did all fix, then populations at $M=0.1$ would adapt much more quickly. 
However, this calculation supports our explanation above:
it appears that adaptation at $M=0.1$ is slower because it must wait for intralocus recombination to free up genetic variation present but masked in the marine population.

\subsection*{Simulation results align with theoretical expectations}

\revpoint{2}{14}
How do our results compare to what is expected from population genetics theory?
Our simulations included many loci under selection and in linkage to many other selected alleles,
which makes precise calculation impossible.
Nonetheless, rough calculations based on simple population genetics theory 
-- with the benefit of hindsight --
turn out to describe qualitatively most of the aspects of adaptation we observed above \citep{Gillespie_1998}. \llabel{Gil_cite}
Because we model stabilizing selection on an additive trait controlled by a moderately large number of loci within each population, more precise expectations might be obtained through quantitative genetics \citep{svardal2014general} or even Fisher's geometric model \citep{barton2001hybridization,chevin2014niche},
but doing so is beyond the scope of this paper

One of the most useful things that theory tells us is about
the fate of a new allele in a lake, that has appeared by either migration or mutation. 
If the allele has fitness advantage $s$ -- \revpoint{3}{8}
i.e., when it is rare but present in $n$ copies,
the expected number of copies in the next generation is $(1+s)n$ --
then the probability that it escapes demographic stochasticity to become common in the population 
is approximately $2s$ \citep{lambert2006probability,haldane1927mathematical} 
, assuming Poisson reproduction, as we roughly have here. 
Since we are studying a quantitative trait, the fitness effect of each allele
depends on the population context:
if the individuals in the population all have trait values $z$ units above the optimum,
and the allele has effect size $-u$ in heterozygotes,
then the fitness advantage of the allele is the ratio of fitnesses with and without the allele.
As we calculate fitness here, this is
$s(u) = \exp(-\beta((z - u)^2) / \exp( - \beta z^2)) \approx 2 \beta z u$,
where $\beta = 1 / 450$. 
This tells us two things: (1) the rate of adaptation decreases as the population approaches the optimum, and (2) larger mutations (in the right direction) are more likely to fix.

\paragraph{New mutations}
The total rate of appearance of new mutations per lake is $\mu_L = 0.04$ per generation, 
and these are divided evenly in six categories: additive, dominant, and recessive, in either direction. 
Therefore, a new additive or dominant effect mutation appears once every 75 generations, on average. 
The effect size of each new mutation is randomly drawn from an Exponential distribution with mean $1/2$, 
and so averaging the probability of establishment over this distribution,
we get that the probability that a dominant mutation manages to establish in a population differing from the optimum by $z$ 
is roughly $\int_0^\infty 4 \beta z u \exp(-2u) du = \beta z$.
Multiplying the rate of appearance of these mutations with the probability they establish
predicts that the rate of establishment of dominant mutations is $\beta z / 75$, 
i.e., about one such mutation every $33750/z$ generations. 
% At the beginning of the simulation where $z = 10$, we would then expect the fixation of alleles to take around 3,000 generations. 
During the initial phase of adaptation, the populations begin at around distance $z=10$ from the optimum,
and $z$ decreases as adaptation progresses.
Mutations that successfully established are more likely to be strong:
the distribution of the effect sizes of these successfully established mutations 
has density proportional to $u \exp(-2u)$,
which is a Gamma distribution with mean 1 and shape parameter 2. 
There are also additive alleles: these have half the effect in heterozygotes, 
and so roughly half the probability of establishment. 
Combining these facts,
we expect adaptive alleles to appear through mutation within lakes 
at first on a time scale of 3,000 generations, 
with the time between local fixation of new alleles increasing as adaptation progresses, 
and each to move the trait by a distance of order 1. 
This agrees roughly with what we see in Figures~\ref{fig:phenotype_time} 
and \ref{fig:phenotype_time_supp}.

\paragraph{Standing variation}
How are freshwater alleles maintained in the ocean, where they are deleterious?
Assuming that fish in the marine environment are close to their phenotypic optimum,
an allele that when heterozygous moves the trait $u$ units in the freshwater direction,
has fitness roughly $\exp(-\beta u^2) \approx 1 - \beta u^2$,
i.e, a fitness differential of $s = \beta u^2$. 
The product of population size and fitness differential in the marine environment for a mutation with $u=1$ is therefore $2Ns \approx 22$, 
implying that these alleles are strongly selected against but might drift to moderate frequency if recessive. 
The average frequency of such an allele in the marine environment at migration-selection equilibrium 
is equal to the proportion of individuals in the ocean replaced by migrants per generation divided by the selective disadvantage, 
i.e., around $m / \beta u^2$. 
Each new lake is likely to contain a few copies of alleles at frequency above 1/400 (since each new lake is initialized with 400 randomly selected genomes). 
For a typical allele with effect size $u=1$, 
the equilibrium migration-selection frequency is greater than this threshold if $m/\beta > 1/400$.
Since $1/\beta = 450$, if $m \ge 5 \times 10^{-6}$ this suggests that there is a good chance that 
any particular lake-adapted allele that is present in all pre-existing lakes 
will appear at least once in the fish that colonize a new lake. 
However, an allele with effect size $u=1$ only has probability of around 1/20 of establishing locally, 
and so must be present in about 20 copies to ensure establishment.
Putting these calculations together, 
we expect migration-selection balance to maintain sufficient genetic variation for new lakes to adapt if $m \ge 10^{-4}$, which corresponds to $M \ge 0.4$.
This is in good agreement with our observations.
% This corresponds to our highest two migration rates, as in e.g., Figure \ref{fig:phenotype_ts2}B.
However, this calculation treats each allele independently;
in practice we found that standing freshwater variation in the ocean
was masked by linkage to compensatory marine variants. \revpoint{2}{15}

\paragraph{Migration}
The key quantity regulating the amount of standing variation in the ocean is the \emph{downstream} migration rate, from lakes to the ocean. 
How important is the upstream migration rate?
If sufficient genetic variation is not present in a new lake initially, it must appear either by new mutation or by migration. 
Since a proportion $m$ of each lake is composed of migrants each generation, 
it takes $1/m$ generations until the genetic variation introduced by migrants equals the amount initially present at colonization. 
This implies a dichotomy: either 
(a) migration is high, and adaptation is possible using variants present at colonization or arriving shortly thereafter, or 
(b) migration is low, so adaptation takes many multiples of $1/m$ generations. 
Since in our model lower migration also reduces the amount of variation available in the ocean, 
we expect very little contribution of subsequent migration across any value of $m$, as seen in Figure~\ref{fig:Origin}.
It remains unknown how different colonization schemes would impact the origin of adapted alleles, 
but it stands to reason based upon our results that early arriving alleles are favored as adaptive alleles in any scenario. \llabel{disc_colon}

\paragraph{Population size}
In our simulations, \revpoint{2}{16}
the marine and freshwater populations are of equal size.
More generally, we expect the downstream migration rate that is required to maintain standing variation there
to scale with the size of the marine population,
since the same number of migrants make up a smaller proportion of a larger population.
So, if the marine population was well-mixed and tenfold larger than all freshwater populations combined,
the number of migrants per lake per generation would need to be ten times larger than in our simulations to maintain sufficient standing genetic variation.
If the marine population was a thousandfold larger,
adaptation from standing genetic variation that is deleterious in the ocean might be impossible. 
The fact that repeated adaptation from standing genetic variation has been shown to occur in real world threespine stickleback populations argues that the population sizes of marine and combined freshwater stickleback are not that different. 
While this might at first seem implausible given the ocean is large and lakes are small, in reality there are hundreds of thousands if not millions of lakes inhabited by threespine stickleback in coastal regions of the Northern Hemisphere, and the limiting habitat in the ocean constraining marine stickleback population sizes is likely to be coastal nesting areas. 
Together, these facts make the equivalency of marine and freshwater stickleback population sizes much more plausible.


\section*{Conclusion}

In this paper, we have analyzed a relatively realistic simulation study of a coastal meta-population to 
understand how repeated local adaptation operates on complex standing variation across a wide range of gene flow rates.
We have shown that rapid and parallel adaptation similar to that documented in natural stickleback populations
occurs over a realistic range of parameter values.
These simulations therefore provide an in-depth look at (and quantitative proof-of-concept of)
the ``transporter'' hypothesis suggested by \citet{schluter2009genetics}.
Selection is able to rebuild the freshwater haplotype at a rapid pace (in tens of generations) from variation present at migration-selection balance in marine populations 
A finding of practical consequence is that the efficacy of $F_{ST}$-based genomic scans varies significantly with migration rate.
Perhaps surprisingly, we found that the majority of adaptive variant can be traced back to the original generation of new habitats,
with little role for post-colonization migrant alleles. 
In other words, a randomly chosen subset of marine stickleback carry the capacity to rapidly adapt in freshwater habitats without \emph{any} continued connection to the rest of of the species.
Thus, the large marine population was able to harbor and distribute alleles that were deleterious in the ocean habitat, but adaptive in freshwater habitats, to surrounding freshwater populations with only a few migrants per population, per generation. 
% Our simulation findings significantly extend previous empirical and theoretical population genomic studies of stickleback to make testable predictions, and provide a framework for similar studies in other organisms to determine the generality of our findings.


\bibliographystyle{plainnat}
\bibliography{Citations}{}

\clearpage
\appendix
\setcounter{table}{0}
\renewcommand{\thetable}{S\arabic{table}}
\setcounter{figure}{0}
\renewcommand{\thefigure}{S\arabic{figure}}

\section*{Supplementary material}

\begin{figure}
    \begin{center}
          \includegraphics{Final_Plots/True_Power_0_25_500.pdf}
          \caption{ 
        Statistical power and true positive rate as a function of $F_{ST}$ threshold. 
        Statistical power is the likelihood that a SNP will be predicted to have an effect on phenotype when there is an effect to be detected.
        True positive rate is the proportion of SNPs above the $F_{ST}$ threshold that affect phenotype.
          \label{fig:Power_FP}
          }
    \end{center}
\end{figure}

\begin{figure}
    \begin{center}
        \includegraphics{Final_Plots/True_Power_0_5_500.pdf}
          \caption{ 
        Statistical power and true positive rate as a function of $F_{ST}$ threshold. As in Figure \ref{fig:Power_FP}
        but computed only using the first one-fifth of the total range -- i.e., fish in the first five lakes and those in the ocean with coordinate less than 5.0.
         	}
          \label{fig:Power_FP_5lakes}
    \end{center}
\end{figure}

\begin{comment}
\begin{figure}
    \begin{center}
          \includegraphics[width=1.0\linewidth]{Final_Plots/Haplo_small.pdf}
          \caption{
            Phenotypes (\textbf{Top}) (effect size and mutation type) and haplotypes \textbf{(Bottom)} as a function of spatial position of the individual 
            possessing the genome. The phenotype of each haplotype is, by definition, the sum of the 
            effect sizes of each mutation in the genome. 
        }
          \label{fig:Haplo_Pheno}
    \end{center}
\end{figure}
\end{comment}


\begin{figure}
    \begin{center}
          \includegraphics[width=1.0\linewidth]{Final_Plots/Pheno_Time_supp.pdf}
          \caption{
        Mean individual trait values in the marine habitat (red line),
                the original lakes (tan lines; average thicker),
                and the new lakes (black lines; average thicker),
                across the course of two simulations, with migration rates of
                \textbf{(A)} $m=5 \times 10^{-3}$,
                \textbf{(B)} $m=5 \times 10^{-2}$ and
        \textbf{(B)} $m=5 \times 10^{-1}$.
                (i.e., $M=1$, 10, and 100 migrants per lake per generation, respectively).
                Optimal trait values in the two habitats are at $\pm 10$.
        }
          \label{fig:phenotype_time_supp}
    \end{center}
\end{figure}

\begin{figure}
    \begin{center}
          \includegraphics[width=1.0\linewidth]{Final_Plots/Fst_Genome_faa0_5_500.pdf}
          \caption{
        Average $F_{ST}$ in windows of 500Mb between:
                \textbf{(top)} marine habitat and old lakes;
                \textbf{(middle)} marine habitat and new lakes; and
                \textbf{(bottom)} old and new lakes.
                Each plot shows $F_{ST}$ values for a separate simulation,
                with columns corresponding to increasing gene flow from left to right.
        $F_{st}$ is calculated between all marine individuals with spatial location $\le 5.0$, and the five corresponding lakes.
        All locations of pre-existing freshwater adapted alleles have been highlighted by 
        partially transparent pink dashed vertical lines, so darker shades of blue imply more 
        freshwater adapted alleles at that location.
        }
          \label{fig:fst_5lakes}
    \end{center}
\end{figure}

%%%%%%% reviews
\newpage

\includereviews
 
\end{document}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% Below is some content that may be useful, but we can delete once the manuscript is finalized.

% Discussion

While our findings prove useful to the understanding of stickleback, they are not autonomous to this species. 
Now, we will discuss the broader implications of our findings on the more general framework of similar systems, we also explore what other questions remain. 
Small patches of similar selective pressures can be found in many places such as high altitudes, underwater volcanic vents, or post forest fire environments. 
Recently,  \citep{Ferretti2018}, discovered seven new species of Hapalotremus (Theraphosinae), living at the extremely high altitudes distributed along the Andes and Yungas in western South America.
With the ancestral species at much lower altitudes, one could image high altitude alleles that could potentially be carried to help in the colonization of neighboring mountain environments. 
Systems of meta-populations like this could potentially arise in many places
where the ancestral species carries around with it beneficial alleles for selective pressures it has encountered in the past and remains connected to through hybridization events.

 \paragraph{Atmosphere's ability to carry whoever scotty is beaming up ... }
Perhaps the most notable finding in this study is that large, ancestral populations have the capacity to maintain alleles which are potentially deleterious for an extended amount of time.  
This principle, deriving solely from the properties of sex and mutation, could be applicable to almost any species experiencing the influence of divergent selection with introgression.
And to explore further how this process is carried out exactly, many of the questions lie in understanding which properties of genomic architecture and demographics allow this transportation to be possible. 
In our simulations, the continued gene flow from freshwater populations meant that the alleles were maintained at a constant rate with fixed demographics. 
Two possible questions that might arise from this are:
(1) How do population sizes, demographics, and evolutionary histories impact an ancestral population's capacity for carrying maladaptive alleles in its opposing environment, and 
(2) What role does genomic architecture play in the ability for a haplotype to be rebuilt.

\paragraph{Realized genomic architecture}
Additionally have also shown introgression is beneficial for inferring causative loci from divergence ($F_{st}$) along the genome. 
This is generally because noise of selectively neutral alleles divergence can appear causative when genetic drift causes more 
differences between populations that have little gene flow between them.
It's important to know that in all scenarios, hitchhiking of selectively neutral alleles could also be 
mistaken for being causative as they often display the same amount of divergence.
In general, our study may suggest that diversity scans may not prove as useful in causal loci inference as previously thought.
It is important to note that these summary statics could prove to be very useful if used beyond just the peaks seen when looked at as a function of genomic position. 
Many methods, such as advanced machine learning, have proved to be very good at taking in noisy data and parsing out important information \citet{Schrider2018}.

\paragraph{Implications for the ``Transporter"-hypothesis}
This study was in large, supportive of the ``Transporter"-hypothesis suggested by \citep{schluter2009genetics}.
Provided there is enough dispersal, we found that repeated colonization and adaptation of freshwater stickleback was able to produce beneficial alleles which could then be 
supported by the marine stickleback population through the offspring of hybridization events. 
When new freshwater environments were colonized by the marine stickleback, selection was able to rebuild the freshwater haplotype. 
Shown in Table~\ref{tab:linkage}, we found that a populations capacity to carry maladaptive alleles remains pretty constant over realistic ranges of dispersal, but low rates of gene flow may 
"mask" the alleles through linkage which counteracts the fitness disadvantage. 
This linkage prevents the ability for a sub-population to rapidly select on the beneficial standing genetic variation as recombination acts as a bottleneck to evolution in this situation. 
Additionally, In Figure~\ref{fig:Origin} we found that the genetic basis for local adaptation, when selecting on standing genetic variation, is propagated largely from the initial generation of inhabitants. 
This contrasts the belief that subsequent migration is a necessity in newly established populations ability to rebuild a haplotype.
In other words, this finding suggests that with high probability, a randomly chosen subset from ancestral species in the meta-populations like marine stickleback, 
carry the capacity to rapidly adapt without \emph{any} continued connection to the rest of of the species.


%%%%%%%%%%%%%%%%%%%%%%%%%%%
%DO WE NEED THIS?
\begin{comment}
\subsection*{thresholds}

We have found that too little migration leads to selection upon new mutations in all subpopulations and lakes alike. In contrast, at high migration rates we have seen that migration load limits the ability for species to locally adapt to the selective pressure of their environment. This leads us to consider a window of introgression which allows for the transportation of FAA's without migration load. 

\subsection*{connect results back to real data?}

\paragraph{The adaptive filter?}
RAMBLING THOUGHTS HERE
Since larger effect alleles are more likely to establish,
be it by mutation or migration,
repeated colonization of new freshwater habitats will select for larger alleles,
be it single alleles or haplotypes bound together by an inversion.
However, these are more strongly selected against in the interstitial time.
Being recessive would help with this,
but would also make it more difficult to establish.

\jgg{TODO: effect of smaller than realistic population sizes?}

\jgg{TODO: Talk about the role recombination plays}
This also brings to light the role of recombination in a system of migration selection balance. 
On one hand, a lower recombination rate would allow for the freshwater haplotype to remain "intact" 
within the marine environment, however, without significant introgression the haplotype would quickly get selected against in marine populations. 
On the other hand a higher recombination rate would, in theory, allow freshwater adapted alleles more longevity 
as they would hitchhike with marine adapted haplotypes. 


\paragraph{Modeling assumptions}
Our simulations had much smaller population sizes than real populations.
How might this affect things?
\end{comment}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%



% Theoretical expectations.

The calculation above ignores allele frequencies -- 
even if a population \emph{could} adapt entirely using the best possible haplotypes,
many of these may be lost to drift.
To account for this, we calculated a similar quantity but accounting for the probability of fixation.
Suppose that the alleles present in the initial population of a new lake
have effect sizes $(x_1, \ldots, x_n)$,
and that the probability of fixation of a single allele of effect size $x$ is $p(x)$.
Let $X_i$ be the average per capita effect of the $i^\text{th}$ allele after fixation,
so that $X_i = 0$ if the allele is lost,
and $X_i = 2x_i$ if the allele fixes (ignoring dominance).
Therefore, the expected contribution to the final trait value of allele $i$
is $\E[X_i] = 2 x_i p(x_i)$.
The trait value after the initial round of adaptation is $\sum_i X_i$,
and so the expected trait value is $2 \sum_i x_i p(x_i)$.
Since there are many alleles that contribute more-or-less additively to the trait,
the probability of fixation of a particular allele is not easy to find,
but we can roughly approximate it as follows:
if the current population trait mean $z$,
then an allele that moves the trait a distance $x$ closer to the optimum
has a fitness advantage in heterozygotes of 
$\exp(-\beta((z - x)^2) / \exp( - \beta z^2)) \approx 1 + 2 \beta z x$, 
where in our parameterization, $\beta = 1 / 450$.
Since the probability of fixation of a single allele with relative advantage $s$
is approximately $2s$ \citep{lambert2006probability,haldane1927mathematical},
then with $z=10$ we can use $p(x) \approx 4 x / 45$.
Putting these considerations together,
a rough estimate for the total shift in trait value resulting from selection
on standing variation in the initial population is
$R = \sum_i 8 x_i / 45$,
where the sum is over all trait-affecting alleles.
This approximation should be good if each trait-affecting allele is rare (present in only a few copies),
effects are additive, and most are unlinked.

We computed $R$ to confirm our suspicion that linkage slowed down adaptation
by showing that there is sufficient allelic variation in the initial populations
to complete adaptation across both migration rates (0.1 and 1 migrants per lake per generations),
but not if the haplotypes of each genomic region are assumed to not recombine.

We calculated for each individual in the initial generation of the new lakes
the total contribution to trait values of each of the ten genomic regions 
that containing trait-affecting mutations.
The recombination rate within each of these ten regions is $10^{-2}$,
so these regions act initially more-or-less like single loci.
Even though the trait is additive across loci,
the presence of sufficiently many distinct alleles to attain, in principle,
the optimal freshwater trait value
does not ensure adaptation will actually occur with these alleles.
On the contrary, many of these will likely be lost to drift.

Between $M = 0.1$ and $M = 1.0$ we see a large difference in the total effect sizes of the haplotypes that are expected to fix.
Given these numbers were computed assuming additivity (slight overestimate) and individuals are diploid,
at $M = 0.1$ the mean total genotype expected to fix in the populations puts individuals at a trait value of $\approx -8$ 
after selection on alleles haplotypes in the initial population. 
In turn, the population must wait for novel mutation or subsequent migration to provide the alleles for the remaining 2 units before 
attaining a mean average trait value at the optimum (-10).
It follows in Figure \ref{fig:phenotype_ts2}B that the majority time to adaptation is spent within 2 units of mean trait value.
The initial adaptation on selection of standing variation up to that point happens quite rapidly.
This provides evidence that linkage to marine alleles has a major impact in the introduced population's ability 
to select and rebuild a pre-existing haplotype given the standing variation at time of introduction. 



