\documentclass{article}
\usepackage[utf8]{inputenc}
\usepackage[english]{babel}
\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{setspace}
\usepackage{natbib} 
\usepackage{graphicx}
\usepackage{subfig}
\usepackage{comment}
\usepackage[backgroundcolor=pink,linecolor=red]{todonotes}
\usepackage{fullpage}
\usepackage[hidelinks]{hyperref}
\usepackage{xcolor}


%\singlespacing
\onehalfspacing
%\doublespacing

\newcommand{\plr}[1]{\todo[linecolor=blue,backgroundcolor=blue!25,bordercolor=blue]{#1}}
\newcommand{\jgg}[1]{\todo[linecolor=green,backgroundcolor=green!25,bordercolor=black]{#1}}
\newcommand{\bill}[1]{\todo[linecolor=green,backgroundcolor=yellow!25,bordercolor=black]{#1}}

\begin{document}

\title{A few stickleback suffice to transport adaptive alleles to new lakes}
\author{Jared Galloway, William A. Cresko, and Peter Ralph}
\maketitle


\section*{Abstract}

Threespine stickleback fish provide a striking example of local adaptation despite recurrent gene flow.
The species is distributed around the Northern hemisphere in both marine and freshwater habitats.
It is thought that these numerous, smaller freshwater populations
have been established "de novo" from marine fish,
and that a shared freshwater phenotype
is often established using standing genetic variation.
Here we use genealogical simulations to determine the levels of gene flow
that best matches observed patterns of allele sharing among habitats in stickleback, 
and more generally to better understand how gene flow and local adaptation in large metapopulations
determine speed of adaptation and reuse of standing genetic variation.
We find that rapid, repeated adaptation using a shared set of alleles
maintained at low frequency by migration-selection balance
occurs over a realistic range of intermediate rates of gene flow.
Low gene flow leads to slow, independent adaptation of distinct habitats,
whereas high gene flow leads to large migration load.
We do not see evidence for strong effects encouraging genomic clustering of causal alleles
or towards particular dominance coefficients.
$F_{ST}$ scans for adaptive alleles are more likely to succeed with higher rates of gene flow.
The results support existing theory of local adaptation,
and provide a more concrete look at a particular, empirically motivated example.  

\textbf{**We need to be more crystal clear in the abstract about the findings from the paper**}

\section*{Introduction}

% Jared's general notes for formatting
%
% - ``citep'' for ``cite in parentheses''. (and, citations come before the punctuation)
% - in latex, use ` `words' ' to get the open- and closed-double quotes (not the double-quote character)
% - "citet" for "cite, as part of the text"

The canonical model for the genetics of adaptation, first formulated by Fisher in the early part of the 20th century, involves the sequential fixation of new mutations. While it has proven valid in numerous studies in the field and the lab, this model is now rightfully understood as incomplete for many species in nature that have more complicated population structures. A growing number of studies have identified instances of convergent or parallel adaptive evolution that utilize standing genetic variation. What evolutionary processes promote the maintenance of re-use of such genetic polymorphisms? How do these processes interact with diverse geographic contexts, as well as variation in genetic processes such as recombination, to change the likelihood of so-called standing genetic variation (SGV). 

Clues come from population genomic studies in the wild that have recently been powered by advances in sequencing technologies. One such organism is the threespine stickleback. This small fish has long been used for studies of behavior, ecology and evolution in the wild, and has more recently become a model for understanding the genetic basis of adaptive evolution. The ancestral marine form has given rise to millions of independently derived populations in recently de-glaciated regions of the Northern Hemisphere, which have each evolved a similar suite - or 'freshwater syndrome' - of phenotypes despite local differences. This adaptive phenotypic evolution to freshwater can occur in mere millenia, in many cases since the end of the last ice age about 13,00 years ago. Studies of the fossil record also clearly show that invasion of oceanic stickleback and local adaptation in freshwater has occurred repeatedly over millions of years, often involving very similar phenotypic shifts as seen in contemporary populations.
    
A finding from recent stickleback population genomic studies is that the same genomic regions, and in some cases the same haplotypes, are implicated in patterns of repeated evolution. These studies show that divergent stickleback populations are clearly still connected via bi-directional gene flow. This reuse of standing genetic variation motivated empirical studies to ask whether this phenotypic and genomic evolution could occur in decades. The answer is a clear yes. For example, in 1964 the Great Alaskan Earthquake caused an uplift of Middleton island and in turn, introduced a group of freshwater ponds around the perimeter of the island. Quickly inhabited by the surrounding marine population of stickleback, \citet{lescak2015evolution} observed significant phenotypic changes in less than 50 years that appear to be parallel to freshwater stickleback that have been separated for over thousands of years. In freshwater stickleback, the number of lateral plates are reduced and the opercle shapes shows the same expansion of the dorsal region and reduction of the ventral region.

 But how can evolution occur so quickly? The leading hypothesis for stickleback is that rather than acting on new mutations, adaptation to freshwater environments is accelerated by selection on SGV found in marine populations. The first clear example of the global reuse of SGV was the gene \textit{eda} which has been shown to be an important regulator for the number of lateral plates. While the low lateral plate version of this gene arose millions of years ago, it is found in freshwater ponds which have formed much more recently. More recently, population genomic studies employing genome-wide haplotype analyses has provided evidence that most regions of the genome that distinguish marine-freshwater genetic differences share this pattern \citep{nelson2017ancient}. Genomic regions that exhibit local adaptation, as evidenced by increased relative population genomic divergence, also show much older haplotypes that are enriched in the alternative habitats. 
 
These empirical data generally support the ``transporter''-hypothesis proposed by Conte and Schluter in 2009 \citet{schluter2009genetics}, which is a conceptual model for the flow of freshwater alleles from multiple smaller freshwater populations into much larger and less structured marine populations through hybridization events. Alleles in this asymetrically structured metapopulation can then be recycled for subsequent freshwater adaptation. Several questions remain, however, about the manner and degree to which these alleles are scattered among the marine fish and must be re-assembled. Is it more akin to the atomization and rebuilding that occurred in the Star Trek series that motivated this hypothesis? Alternatively does it occur in more of a patchwork by individuals, or their early generation hybrids, in geographically adjacent freshwater habitats moving quickly through the marine environment to seed new freshwater populations. More generally, how do different levels of gene flow and local adaptation interact with the asymmetric nature of stickleback population structure affect the dynamics of the origin and rate of adaptive alleles, as well as their eventual organization into marine and freshwater typical genomes? 

Here we develop a forward simulation approach in SLiM to model the stickleback population structure in marine and freshwater habitats to document the effect of variation in gene flow and recombination on the genetic and genomic architecture of local adaptation. We address how rapidly can selection act on standing genetic variants at a given value of migration, $M$, and what the precise origin is of those variants. We ask what scale of historical introgression between populations with selective pressures $X$ and $Y$, allows for rapid and parallel local adaptation of a population derived from $X$ to an introduced environment with selective pressure $Y$? How are the variants structured, historically and across the genome underlying the trait? Furthermore, how can we infer causal loci for regions of the genome that must be driving the rapid adaptation, from real data. Many biologists today make use of genome wide association studies (GWAS) and $F_{st}$ across the genome to estimate regions responsible for certain traits. Unfortunately, the data can be heavily influenced by the level of introgression and population structure of the samples. This being the case, what are the biases we can expect to see in real data for certain levels of introgression. 

We find that .... \textbf{1. rewrite the previous paragraph to be more concise and tight, and 2. briefly restate our major findings which should line up well with those presented in the abstract. The following I grabbed from the beginning of the Results section}
*****
We varied the migration rate, $m$, from $5 \times 10^{-5}$ to $5 \times 10^{-2}$ per individual, i.e., between 0.01 and 10 migrants per year to and from each of the ten lakes. This had a strong effect on many aspects of adaptation, including how fast adaptation occurred in each lake, how much alleles were shared between lakes,
and the population genetic signals left behind. At the lowest migration rate, lakes adapted nearly independently, while at the highest migration rate, the habitats are beginning to act like a single metapopulation with substantial migration load. We will now dissect what happens between these extremes. Interestingly, the ability for separate populations to locally adapt to their own selective pressure was relatively unaffected until the highest rate of migration between the marine and freshwater environments. All rates of historical introgression aside from the lowest, helped both the initial and introduced freshwater populations share pre-existing freshwater adapted alleles. The sharing of alleles resulted in rapid adaptation ($\approx 100$ generations) of the introduced population (split from the marine population) to adapt to the freshwater environment. We also found that larger amounts of migration allow for more statistical power and less false positives in the resulting population genetics data ($F_{st}$ per SNP across the genome).
****


%%%%%%%%%%%%%%%%
\section*{Methods}

We explored these questions using forwards-time simulations with explicit genomic representation of a quantitative trait in SLiM \citep{haller2017slim,haller2018slim3}.
The details of the model were motivated by current understanding of threespine stickleback history and demography,
but remain simplistic in some aspects due to computational constraints.
Possibly the most important caveat
is that simulated population sizes are much smaller than the census size of the threespine stickleback population
(see the Discussion for more on this).

\paragraph{Habitat and geography}
Each habitat type -- marine and freshwater -- have a fixed total of 5,000 diploid individuals each. 
The arrangement of these habitats, depicted in Figure \ref{fig:Geo},
roughly models a set of freshwater habitats along a stretch of coastline. 
The marine habitat  is a continuous, one-dimensional range of 25 units of length,
while the freshwater habitat is divided into ten subpopulations (which we call ``lakes''),
each connected to the marine habitat at regularly spaced intervals
(positions $i - 1/2$ for $1 \le i \le 25$).

Divergent selection is mediated by a single quantitative trait
with different optima in marine and freshwater habitats.
This situation roughly models the cumulative effect of the various phenotypes such as armor morphology, 
body size, craniofacial variation and opercle shape on which divergent selection is thought to act
in the two environments. 
Concretely, the optimal trait values in the marine and freshwater habitats are $+10$ and $-10$ respectively,
and fitness of a fish with trait value $x$ in a habitat with optimal value $x_\text{opt}$
is determined by a Gaussian kernel with standard deviation 15, i.e.,
\begin{align*}
    f(x; x_\text{opt})
    &=
    \exp\left\{
        \frac{1}{2}
            \left(
            \frac{x-x_\text{opt}}{15}
            \right)^2
        \right\} .
\end{align*}
Note that only the scale of trait values relative to mutation effect size is relevant.
We chose the difference between optima and strength of stabilizing selection in each habitat
so that 
(a) around 10 (diploid, homozygous) mutations were sufficient to move from one optimum to the other,
and (b) well-adapted fish from one habitat would have low, but nonzero, fitness in the other habitat.

\begin{figure}
	\begin{center}
  		\includegraphics[width=0.6\linewidth]{GeographyFigure.pdf}
  		\caption{
            \textbf{Diagram of simulated populations:}
            a single, continuous, one-dimensional marine habitat (blue)
            is coupled to randomly mating ``lakes'' at discrete locations.
            After an initial period of 100K generations with 25 lakes,
            an additional 25 lakes are added (at the same set of locations)
            to simulate the appearance of newly accessible freshwater habitats.
            The marine habitat, and each set of 25 lakes, each contain 5,000 individuals at all times.
			}
  		\label{fig:Geo}
	\end{center}
\end{figure}


\paragraph{Genetic architecture of the trait}
Each individual carries two copies of a linear chromosome of $10^8$ loci.
Mutations that can affect the trait under selection can occur at rate $10^{-7}$ per locus per generation
in ten ``effect regions'' of $10^5$ loci each,
spread evenly along the chromosome.
Each mutation in these regions is either additive, completely recessive, or completely dominant (with equal probability), 
and with an effect sizes chosen randomly from an Exponential distribution with mean $1/2$, either positive or negative with equal probability. 
Individual trait values are determined additively from the diploid genotypes. 
Concretely, an individual that is heterozygous and homozygous for mutations at sets of loci $H$ and $D$ respectively has trait value 
$x = \sum_{i \in H} h_i s_i + \sum_{j \in D} s_j$, 
where $h_i$ and $s_i$ are the dominance coefficient and the effect size of the mutation at locus $i$.
Subsequent mutations at the same locus replace the previous allele.


\paragraph{Population dynamics}
We use SLiM to simulate a Wright--Fisher population with nonoverlapping generations 
and a fixed population size of 5000 diploid individuals in each habitat.
Each generation, the two parents of each new offspring are chosen proportional to their fitness 
(all individuals are hermaphroditic),
and the contributing genomes are produced by Poisson recombination 
with an average of one crossover per chromosome per generation 
($10^{-7}$ per locus per generation). 
Since the total population across \emph{all} lakes is regulated,
to keep population sizes roughly constant within each lake, 
before selection we divide fitness values of each freshwater individual 
by the mean fitness in their lake,
so that the mean fitnesses of all lakes are equal.

Dispersal occurs both locally along the coastline in the marine habitat 
and between the marine habitat and the lakes, and can be thought of as occurring at the juvenile stage. 
There is no dispersal directly between lakes.
The lake--ocean migration rate is denoted $m$, 
which we refer to as the rate of \emph{gene flow} between habitats.
Each new individual in the marine habitat has freshwater parents with probability $m$; 
to obtain the pair, a first parent is chosen proportional to fitness, and a mate is chosen from the same lake as the first, also proportional to fitness. 
The resulting offspring is given a spatial location in the marine habitat at the location of the parent's lake. 
Parents for a new marine individual who is not a migrant are chosen similarly (with probability $1-m$): 
first, a single parent is chosen proportionally to fitness in the marine habitat,
and then a mate is chosen, also proportionally to fitness but re-weighted by a Gaussian function of the distance separating the two, with standard deviation $1/2$. 
Concretely, if the first parent is marine individual $i$, then marine individual $j$ is chosen as the mate with probability proportional to $f(x_j) \exp(-2d_{ij}^2)$,
where $d_{ij}$ is the distance between the two locations. 
Finally, each new marine offspring is given a position displaced from the first parent's position by a random Gaussian distance with mean 0 and standard deviation 0.02, 
and reflected to stay within the population.
New offspring in the freshwater habitat are chosen in the same way, except the probability that the parents are marine individuals is $m$; 
any new freshwater offpsring produced by marine individuals are assigned to the lake nearest to the position of the first marine parent.


\paragraph{New freshwater populations} 
To study how newly appearing freshwater habitats adapt, we introduce a new set of 25 lakes midway through the simulation.
The initial set of individuals in these new lakes have parents chosen in the same way from the marine habitat as ocean-to-lake migrants,
and act like an independent copy of the original set of lakes -- in particular, 
the two sets of lakes each have 5,000 individuals at all times. 
(Since this doubles the number of lake-to-marine immigrants, 
after this happens the probability that a new marine individual has freshwater parents is $2m$ instead of $m$.)

We quantify genetic differentiation between the habitats with per-locus $F_{ST}$:
if $p_f$ and $p_m$ are the frequencies of a given mutant allele in the freshwater and marine habitats,
respectively,
and $\bar p = (p_f + p_m)/2$,
then $F_{ST} = 1 - (p_f (1-p_f) + p_m (1-p_m))/ (2 \bar p (1-\bar p))$.


\paragraph{Recording genealogical history}
We used SLiM's ability to record \emph{tree sequences} \citep{haller2018treesequence}
to record the genealogical history of all individuals alive 
at the time of introduction of new lakes, the time of adaptation, and the end of the simulation.
This allowed us to directly query the true origins of adaptive alleles,
as well as allowing much larger simulations
by avoiding the computationally expensive task of simulating neutral mutations
(which were added after the fact at a rate of $10^{-7}$ per locus per generation, 
as described in \citet{kelleher2018efficient}).

The output tree sequence from each simulation allows us to,
at each locus along the genome,
construct the genealogical tree relating all extant chromosomes
as well as all chromosomes present at time of introduction of the new lakes.
Using these trees,
we classified each freshwater allele found in the new lakes at time of adaptation
into four categories:
\begin{enumerate}
    \item \emph{De novo alleles:}
        deriving from a new mutation that occurred in a new lake.
    \item \emph{Migrant alleles:} 
        not present in the individuals that initially colonized the new lake, 
        but brought in by migration at a later time.
    \item \emph{Captured alleles:}
        present in the individuals that initially colonized the new lake, 
        and both common (above 50\%) in the original lakes,
        and uncommon (below 50\%) in the ocean.
    \item \emph{Marine alleles:} 
        present in the individuals that initially colonized the new lake, 
        and not a captured allele.
\end{enumerate}
Roughly speaking, these four categories describe the contribution of
(1) new mutation, 
(2) post-colonization migration, 
(3) standing variation at migration--selection balance, and
(4) standing variation at mutation--selection balance.
We could partition migrant alleles into finer categories
(e.g., we do not distinguish whether they are common in the original lakes or not),
but this category turns out to be small.

%% we did this, but it is not used in the results
% We also used the tree sequence to measure the realized fitness of individuals 
% in the initial generation of the new lakes,
% measured as
% the total number of eventual offspring, in the generation at the time of adaptation, 
% which have inherited a adaptive variant from that individual. 
% To calculate this, we only considered gene trees which overlapped an adaptive variant site (defined at time of adaptation).
% We then counted the total number of offspring which had inherited from any adaptive variant site, 
% for each individual in the initial generation. 

We were also able to use the tree sequence to get information about the individual ``effect regions''
in all initial genomes of the new lakes at time of introduction.
From this we determined the ability for the new lakes to select upon the standing variation in the marine.
Given the probability that effect region fixes at a given trait value, $2 * x / 45$, where $x$ is the effect size of a haplotype,
we calculated the expected total effect sizes of the haplotypes that fix.
Concretely, it is the sum across the 10 effect regions of
$$\sum_{i=1}^N \prod_{j < i} (1 - p_j) p_i x_i$$
where $N$ is the number of genomes per lake, $p_i$ is the probability of fixation of the 
$i^{th}$ haplotype, $x_{i}$ is the effect size of the $i^{th}$ haplotype, and the haplotypes are sorted in decreasing order by $x_{i}$.
These numbers were computed assuming additivity of mutations, meaning the numbers produced are a slight overestimate.

\paragraph{Descriptive statistics}\bill{Population genetic analyses? Instead of the more general 'Descriptive statistics'?}
To assess whether new lakes adapt using existing genetic diversity,
we define a \emph{freshwater allele}
to be an effect mutation that has frequency higher than $0.5$ in at least one of the original lakes,
while remaining lower than $0.5$ in the marine. 
This categorization is made for each generation using the allele frequencies from that generation,
and so changes with time.
Alleles common in the newly introduced lakes do not count
if they are not also common in the original lakes.
They are defined this way because the transportation hypothesis 
does not specify where or when an advantageous mutation arises,
but simply suggests that any sufficiently common freshwater adapted allele 
could participate in adaptation in new habitat \citep{schluter2009genetics}.

\emph{Time to adaptation} of the introduced population, denoted $T_\text{adapt}$, 
is defined to be the generation at which 
the difference between the average trait value in the original and the introduced freshwater populations 
is less than 0.5. 

We describe overall genetic differentiation between the habitats using $F_{ST}$,
calculated on a per-locus basis.
Concretely, if $p_f$ and $p_m$ are the frequencies of a given mutant allele
in the freshwater and marine habitats, respectively,
and $\bar p = (p_f + p_m)/2$,
then we compute $F_{ST}$ for that mutation as $1 - p_f p_m / (\bar p (1-\bar p))$.

%%%%%%%%%%%%%%%%%%
\section*{Results}

We varied the ocean--lake migration rate, $m$, 
across separate simulations
from $5 \times 10^{-5}$ to $5 \times 10^{-1}$.
Since each lake contains 200 individuals,
this corresponds to between 0.01 and 100 migrants per lake per generation.
Many aspects of adaptation changed substantially across this range,
including the speed of adaptation,
degree of sharing of adaptive alleles between lakes,
and the population genetic signals left behind.
At very low rates of gene flow,
each new lake's population adapted \emph{de novo},
which took a very long time.
At very high rates of gene flow, local adaptation was almost impossible.
Between these two extremes,
genetic variation that allowed adaptation to freshwater habitats
could move relatively easily between lakes --
despite being deleterious in the intervening marine habitat --
which allowed populations arriving in new lakes to adapt quickly,
reassembling collections of alleles responsible for freshwater adaptation
in other lakes.


%%%%%%%%%%%%
\subsection*{Local Adaptation: differentiation with gene flow}

\begin{figure}
	\begin{center}
        \includegraphics{Final_Plots/Pheno_Time.pdf}
  		\caption{ 
        		Mean individual trait values in the marine habitat (blue line),
        		the original lakes (yellow lines; average in orange),
        		and the new lakes (light green lines; average in dark green),
        		across the course of two simulations, with migration rates of
        		\textbf{(top)} $m=5 \times 10^{-5}$ and
        		\textbf{(bottom)} $m=5 \times 10^{-4}$ per generation per individual
        		(i.e., 0.01 and 0.1 migrants per lake per generation, respectively).
        		Optimal trait values in the two habitats are at $\pm 10$.
                Analogous plots for other migration rates
                are shown in Figures \ref{fig:pheno_by_time1} and \ref{fig:pheno_by_time2}.
		}
  		\label{fig:phenotype_ts2}
	\end{center}
\end{figure}

\begin{figure}
	\begin{center}
  		\includegraphics{Final_Plots/Pheno_Dist.pdf}
  		\caption{
            Distribution of mean trait values across generations of the simulation,
            for different migration rates.
            The dashed pink and purple lines at $\pm 10$ give the optimum phenotypes
            in the marine and freshwater environments, respectively.
		}
  		\label{fig:MeanPhenotype}
	\end{center}
\end{figure}

Local adaptation occurred at most migration rates,
as shown in Figure~\ref{fig:MeanPhenotype}.
At the highest migration rate (at which half of each population was composed of migrants),
populations had only slightly different average trait values.
At lower migration rates (10 migrants per generation and below),
populations adapted to local conditions:
as shown in Figures \ref{fig:phenotype_ts2},
\ref{fig:pheno_by_time1} and \ref{fig:pheno_by_time2},
freshwater and marine populations diverged
until the trait means were close to the optimal values in each habitat. 
The establishment of new alleles in the lakes is visible in Figure \ref{fig:phenotype_ts2}
as jumps in the mean trait value,
which move the trait by an amount of order 1 every few hundred generations.
Trait variation within each population was small compared to the difference between populations.
Across all parameter values, differences at around 16 commonly polymorphic sites 
(eight that shift the trait in each direction)
were responsible for most of the adaptive differences between freshwater and marine habitats.

As expected, increasing migration rate decreased differentiation between habitats. 
As seen in Figure~\ref{fig:Fst}, 
$F_{ST}$ between marine and freshwater habitats 
at neutral sites steadily declines as migration increases. 
However, local adaptation was still able to occur despite overall homogenization 
at up to 10 migrants per lake per generation.
% if computed using only sites with alleles affecting the trait (``effect mutations''),
% $F_{ST}$ between habitats was relatively constant across migration values.

\begin{figure}
	\begin{center}
        \includegraphics[width=\textwidth]{Final_Plots/Fst_Genome.pdf}
  		\caption{
		Average $F_{ST}$ in windows of 500Mb between:
        \textbf{(top)} marine habitat and old lakes;
        \textbf{(middle)} marine habitat and new lakes; and
        \textbf{(bottom)} old and new lakes.
        Each plot shows $F_{ST}$ values for a separate simulation,
        with columns corresponding to increasing gene flow from left to right.
		Ten vertical dotted pink lines in each subplot 
        show regions which have the potential to affect phenotype.
     } \label{fig:Fst}
	\end{center}
\end{figure}

\paragraph*{Speed of adaptation}
Adaptation occurred much more quickly
at higher migration rates,
both in the old and new sets of lakes.
We measured this ``time to adaptation''
as the number of generations until 
average trait values in old and new lakes were within 0.5 of each other,
shown in Figure \ref{fig:TimeToAdaptation} for different rates of gene flow.
Adaptation of new lakes took over 18,000 generations at the lowest rate,
while at one migrant per lake per generation
new lakes managed to adapt in only 32 generations. 

\begin{figure}
	\begin{center}
  		\includegraphics{Final_Plots/Time_Adapt.pdf}
  		\caption{
		Time to adaptation as a function of migration rate.
        The time to adaptation is measured as the number of generations until
		the introduced population's mean phenotype 
        comes within 0.5 of the original lakes average phenotype. 
		Each point represents a single simulation run,
		and the yellow dashed line is the average of all points at each respective parameter value.
        (Adaptation did not occur at the highest rate of gene flow.)
        \textbf{where are the multiple points?}
        } \label{fig:TimeToAdaptation}
	\end{center}
\end{figure}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Sharing of freshwater adapted alleles}

%FIGURES THAT SHOW THIS:
%-MPAA / IND
At low migration rates, the \emph{initial} period of adaptation
takes roughly ten times longer for lakes as it does for the ocean.
This difference occurs because at low migration rates, 
adaptation occurs independently in each lake,
and the marine habitat has ten times as many individuals,
and therefore ten times the influx of new alleles than any one lake. 
At higher migration rates, greater mixing
allows lakes to share alleles instead of developing their own genetic basis of adaptation. 
As a first indication of this, Figure \ref{fig:Fst} 
shows that $F_{ST}$ between the ``original'' and ``introduced'' sets of lakes 
at effect mutations decreased with migration rate.

To investigate in more depth how locally adaptive alleles found in the original lakes
are shared between lakes, and how they spread to the new lakes,
we counted ``pre-existing freshwater adapted alleles''.
These we define for a particular generation 
to be any mutation with an effect on the trait
whose frequency is above 50\% in at least one original lake 
and below 50\% in the marine habitat.
Figure \ref{fig:MPFAI}A shows the distribution of the number of these alleles, 
across generations.
At the lowest migration rate (one migrant per 100 generations per lake), 
each lake has a private set of about 10 mutations
nearly fixed in that lake but not elsewhere:
new lakes acquire new adaptive alleles and so have none of these.
At one migrant per 10 generations per lake,
the original set of lakes each developed private alleles,
but these are present at low frequency in the ocean,
and so new lakes adapt 
mostly using a subset of this large repertoire of standing variation.
As migration rate increases past this, 
the number of pre-existing freshwater adapted alleles declines
because alleles move between populations by migration
before they can appear by new mutation,
while the frequency of these alleles in the ocean
stays relatively constant.

Figure~\ref{fig:MPFAI}B shows the distribution of 
the mean percentage of currently-defined freshwater adapted alleles 
that each genome in each of the populations carries, 
averaged across time and individuals. 
If all individuals across lakes carried the same set of alleles determining their trait value, 
this would be 100\%. 
At the lowest migration rates,
each genome in the original lakes have almost exactly~$1/25^{th}$ 
of the total number of pre-existing freshwater adapted alleles -- 
this is because each of the 25 lakes has adapted with a unique set of alleles. 
Since these are \emph{pre-existing} alleles, the value is zero for introduced lakes.
Figure \ref{fig:MPFAI}A shows us that at $0.1$ migrants per lake per generation and above,
the average individual across the new lakes has nearly the same amount of 
pre-existing freshwater adapted alleles as individuals across the old lakes.
As expected, the genetic basis of the freshwater phenotype 
seems to simplify as migration increases --
higher rates of migration allow
adaptive alleles of higher effect to travel more efficiently through the population,
even though they are deleterious in the ocean.

The numbers of Figure~\ref{fig:MPFAI} suggest that
the dramatic increase in speed of local adaptation we observed above occurs because
higher gene flow between populations allows sharing of freshwater alleles between populations.
We confirmed this by using recorded tree sequences to identify the origin of each trait-affecting
allele common in the new lakes at time of adaptation,
as defined in the Methods.
Figure~\ref{fig:Origin} shows that at the lowest rate of gene flow,
the majority of adaptive alleles are derived from de novo mutation.
As gene flow increases, a larger fraction of adaptive alleles 
derive from pre-existing variation in the marine population at the time of introduction.
In other words, 
greater mixing at higher migration rates allows lakes to share alleles
instead of developing their own genetic basis of adaptation.

\begin{figure}
	\begin{center}
  		\includegraphics{Final_Plots/Freshwater_Alleles.pdf}
  		\caption{
            Amount of standing freshwater variation by habitat, across migration rates.
            Each plot counts ``pre-existing freshwater adapted alleles'',
            that are common in the original lakes but rare in the ocean
            (see text for definition).
            \textbf{(A)} Mean number of these alleles per individual.
            \textbf{(B)} Mean percentage of these alleles per individual.
            \textbf{(C)} Total number of these alleles (so, $B = A/C$).
            The number of alleles meeting these conditions changes over the course of the simulation,
            and each plot shows distributions of these values across generations.
            The horizontal axis shows $M$, the mean number of migrants per lake per generation.
		}
		\label{fig:MPFAI}
	\end{center}
\end{figure}


\begin{figure}
	\begin{center}
  		\includegraphics[width=0.7\linewidth]{Final_Plots/Allele_Origin.pdf}
  		\caption{ 
        \textbf{(Origin of adaptive alleles:)}
        Each bar plot shows the origins
		of all trait-affecting alleles above frequency 50\% in at least one new lake,
        classified as
		\textbf{(red)} new mutations,
		\textbf{(blue)} post-colonization migrants,
		\textbf{(green)} ``captured'' from pre-existing lakes, or
		\textbf{(orange)} standing marine variation.
        See Methods for precise definitions of these categories.
		}
  		\label{fig:Origin}
	\end{center}
\end{figure}

At first, increased migration allows sharing of adaptive alleles between lakes, 
but at the highest migration rate, 
the constant influx of alleles between the habitats creates substantial migration load. 
The rate at which migration load becomes substantial,
10 migrants per lake per generation,
only replaces 5\% of each population each generation with migrants from the other habitat, 
but this is sufficient to shift the mean trait values to nearly half their optimal values, 
as seen in Figure \ref{fig:MeanPhenotype}. 

\paragraph{Standing genetic variation} 
Despite a dramatic difference in the amount of allelic sharing between lakes, 
standing genetic variance (Figure \ref{fig:SGV}) 
was around 0.05 in the freshwater habitat across all but the highest migration rates. 
Concurrently, genetic variation in the marine population 
steadily increased with migration to a similar value. 
On the one hand, it is not surprising that lakes, as a group, 
carry more genetic variation than the marine habitat, 
since population subdivision allows different alleles to become common in each. 
However, it is striking that at $m=.005$, 
the marine habitat carries as much genetic variation, 
despite a lack of any substantial migration load.
\plr{perhaps we want to remove this bit? revisit if it is telling us something important.}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Realized genetic architecture}

Now we take a closer look at the genomic architecture of local adaptation between the two habitats. 
Do the alleles underlying trait differences cluster along the genome?
Do measures of local differentiation identify the causal loci? 
Figure \ref{fig:Fst} shows plots along the genome of per-locus $F_{ST}$ values between habitats.
(Note that we are pooling freshwater habitats; a single lake would provide substantially less power.)
\plr{need to know where the FAA are in these figures}
Higher migration rates showed more distinct $F_{ST}$ peaks over polymorphic loci 
underlying trait differences between the habitats. 
``Background'' levels of $F_{ST}$ increase as gene flow decreases,
swamping out this signal until the regions under selection are indistinguishable. 
This is likely due to two reasons: 
first, stronger genetic drift with less migration leads to higher background $F_{ST}$, and 
second, greater sharing of adaptive alleles providing a shared signal across populations.

This suggests that genome scans for local adaptation based purely on measures of differentiation 
will only be successful given enough migration between habitats. 
To quantify this, 
Figure \ref{fig:Power_FP} shows the power and false positive rates 
that would be obtained by an $F_{ST}$ cutoff 
that declared everything above a certain value to be a causal locus.

\plr{Need to fix caption on figure \ref{fig:Power_FP} before editing this bit.}

In regions of the genome underlying individual trait value, we observed Given that migration increases the gene flow between subpopulations, how valid are $F_{st}$ peaks at different $M$. Knowing exactly which mutations effect phenotype in our simulations, we can look at the statistical power and false positives given $F_{st}$ per SNP across the genome. In Figure \ref{fig:Power_FP} , looking at an $F_{st}$ threshold greater than 1, we see the two lowest migration rates $10^{-5}$ and $10^{-4}$ having very little statical power. This along with low false positive rate across all $F_{st}$ threshold values is fairly predictable when you consider the high $F_{st}$ values across the genome. 



%%%%%%%%%%%%%%%%%%%%%
\subsection*{Origin of introduced freshwater adaptive alleles}

We have thus far found that the speed of adaptation depends strongly
on the degree to which alleles can be shared between populations.
However, there remains a mystery.
Figure~\ref{fig:MPFAI}A shows that with $m = 0.1$ migrants per lake per generation, 
there is substantial sharing of alleles between populations,
and yet Figure~\ref{fig:TimeToAdaptation} shows that new lakes 
take around 2,000 generations to adapt.
This is surprising because we see a similar quantity of allele sharing at  $m = 1.0$, 
but much more rapid adaptation ($\approx 30$ generations). 
Perhaps
this 60$\times$ discrepancy occurs because 
new lakes must wait for an additional influx of freshwater alleles through migration
beyond what is present in the initial generation?
However, Figure~\ref{fig:Origin} shows that at both $m = 0.1$ and $m = 1.0$,
the vast majority of alleles underlying the local adaptation
derived from individuals in the original generation (``captured'' or ``pre-existing'').

An alternative explanation is that at lower rates of gene flow,
freshwater alleles present as standing variation in the marine habitat
are more tightly linked to marine alleles,
and so adaptation in the new lakes must wait for recombination.
This might be expected since freshwater alleles present in the ocean,
to remain present at migration-selection balance at low migration rates,
must be ``masked'' by nearby compensatory alleles.
Reversing this ``masking'' then slows the process of adaptation that uses this genetic variation.
Said another way,
higher gene flow from freshwater to the ocean
maintains relatively intact freshwater haplotypes
that can be more easily rebuilt.
To confirm this, we calculated for each individual in the initial generation of the new lakes
the total contribution to trait values of each of the ten genomic regions 
that containing trait-affecting mutations.
The recombination rate within each of these ten regions is $10^{-2}$,
so these regions act initially more-or-less like single loci.
Even though the trait is additive across loci,
the presence of sufficiently many distinct alleles to attain, in principle,
the optimal freshwater trait value
does not ensure adaptation will actually occur with these alleles.
On the contrary, many of these will likely be lost to drift.
To get a rough idea of whether the genetic variation present in the initial population
is sufficient for the population to adapt,
we translated the total effect size $x$ of each region 
to a probability of fixation
(by $p(x) \approx 2x/45$),
and then calculated $\sum_i x_i p(x_i)$.
If a region moves the trait an average of $x$ units towards the freshwater optimum
(averaging over dominance relationships),
Given the probability that an ``effect region'' fixes at a given trait value, 
$2 \times x / 45$ where $x$ is the effect size of a haplotype,
we were able to determine expected total effect sizes of the haplotypes that fix (detail about computation in methods).
Below, we see the mean of these expected total effect sizes across all lakes.

\begin{center}
\begin{tabular}{ | c | c  | c  | c | }
 \hline
 $m = 0.01$ & $m = 0.1$ & $m = 1.0$ & $m = 10.0$ \\ 
 $-0.1227$ &  $-4.6976$ & $-12.9298$  & $-12.2710$\\
 \hline
\end{tabular}
\end{center}

Between $m = 0.1$ and $m = 1.0$ we see a large difference in the total effect sizes of the haplotypes that are expected to fix.
Given these numbers were computed assuming additivity (slight overestimate) and individuals are diploid,
at $m = 0.1$ the mean total genotype expected to fix in the populations puts individuals at a trait value of $\approx -8$ 
after selection on alleles haplotypes in the initial population. 
In turn, the population must wait for novel mutation or subsequent migration to provide the alleles for the remaining 2 units before 
attaining a mean average trait value at the optimum (-10).
It follows in Figure \ref{fig:phenotype_ts2}B that the majority time to adaptation is spent within 2 units of mean trait value.
The initial adaptation on selection of standing variation up to that point happens quite rapidly.
This provides evidence that linkage to marine alleles has a major impact in the introduced population's ability 
to select and rebuild a pre-existing haplotype given the standing variation at time of introduction. 

\subsection*{Theoretical expectations}
\plr{does this go first or second?}

Here's some rough calculations to get a sense for what should be going on. Everything is done in more detail in OTHER PLACES WE SHOULD CITE.
Fisher, Chevin, etc.

Suppose a new allele enters a lake, either by migration or mutation. If, when it is rare but present in $n$ copies, it has fitness advantage $s$ -- i.e., the expected number of copies in the next generation is $(1+s)n$ --then the probability that it escapes demographic stochasticity to become common in the population is approximately $2s$ \citep{fisher,prob_fixation}. If the current population all differed from the optimum trait by $z$, and the allele has effect size $-u$ in heterozygotes, then the fitness advantage of the allele would be $s(u) = \exp(-\beta((z - u)^2) / \exp( - \beta z^2)) \approx 2 \beta z u$, where in our parameterization, $\beta = 1 / 450$. This tells us two things: (1) the rate of adaptation decreases as the population approaches the optimum, and (2) larger mutations (in the right direction) are more likely to fix.

\paragraph{New mutations}
The total rate of appearance of new mutations per lake is $\mu_L = 0.04$, which are divided evenly in seven categories: neutral, and then additive, dominant, and recessive in either direction. This implies that a new additive or dominant effect mutation appears once every 87.5 generations, on average. Effect sizes are randomly drawn from an Exponential distribution with mean $1/2$, and so the probability that a dominant mutation manages to establish in a population differing from the optimum by $z$ is roughly $\int_0^\infty 4 \beta z u \exp(-2u) du = \beta z$, and so the rate of establishment of dominant mutations is $\beta z / 87.5$, i.e., about one such mutation every $2461/z$ generations. The distribution of these successfully established mutations  has density proportional to $u \exp(-2u)$, i.e., is Gamma with mean 1 and shape parameter 2. Since additive alleles have half the effect in heterozygotes, they have half the probability of establishment. During the initial phase of adaptation, the populations begin at around distance $z=10$ from the optimum. Combining these facts, we expect adaptive alleles to appear through mutation at first on a time scale of 250 generations, with the time between local fixation of new alleles increasing as adaptation progresses, and each to move the trait by a distance of order 1.

\paragraph{Standing variation}
An allele that moves the trait $z$ units in the freshwater direction in heterozygotes has fitness roughly $\exp(-\beta z^2) \approx 1 - \beta z^2$ in the marine environment (which is close to optimal). The product of population size and fitness differential in the marine environment for a mutation with $z=1$ is therefore $2Ns = 8.9$, implying that these alleles are strongly selected against but might occasionally drift to moderate frequency. The average frequency of such an allele in the marine environment at migration-selection equilibrium is equal to the total influx of alleles per generation divided by the selective disadvantage, which if $M = 2000 m$ is the number of immigrants per generation, is $2 M / \beta z^2$. With $z=1$, the factor multiplying $M$ is $2/\beta z^2 \approx 1/200$: since lakes have 400 genomes, as long as $M \ge 1$, the chances are good that any particular lake-adapted allele that is present in all pre-existing lakes will appear at least once in the fish that colonize a new lake. However, an allele with $z=1$ only has probability of around 1/20 of establishing locally, suggesting that we'd need $M \ge 10$ to ensure enough pre-existing genetic variation that adaptation would happen entirely using the initial set of colonizers. This corresponds to our highest two migration rates, as in e.g., Figure \ref{fig:phenotype_ts2}B.

\paragraph{Migration}
If sufficient genetic variation is not present in a new lake initially, it must appear by new mutation or migration. Since a proportion $m$ of each lake is composed of migrants each generation, it takes $1/m$ generations until the genetic variation introduced by migrants equals the amount initially present at colonization. This implies a dichotomy: either (a) adaptation is possible using variants present at colonization or arriving shortly thereafter, or (b) adaptation takes many multiples of $1/m$ generations.

These calculations depend on there being no bottleneck in colonization of the lake; if there is a bottleneck, then an additional factor must be added. At what point do we expect new mutation to be more important than migration for adaptation? By the calculations above, if $M \ge 10$, we expect initial diversity in a lake to be sufficient  for adaptation, corresponding to our third-highest migration rate. If this does not happen, then we expect adaptation to take a multiple of $1/m$ generations;
with our values, $1/m$ ranges from 20,000 to 20 generations. Above we estimated that adaptive alleles due to new mutation fix locally about every 2,000 generations, suggesting that at our second-lowest migration rate (where $1/m = 2,000$), the two contributions of migration and new mutation are roughly equal.
This is in fact what we see: in Figure \ref{fig:MPFAI}, we see that at the second migration rate, alleles start to be shared between lakes, while by the third migration rate, they are almost entirely shared.

%%%%%%%%%%%%%%%%%%%%
\section*{Discussion}

We have shown that historical introgression, at our given parameter sets, is able to reproduce rapid and parallel adaptation similar to what we've seen in real populations such as Middleton island. Selection is able to rebuild the freshwater haplotype from marine populations as a medium between all freshwater populations. Almost all rates of migration were helpful in the efficiency of the population to locally adapt except for the highest at which migration load limited the ability of the populations to reach the local optimum. 

We also have also explored the genomic architecture as a consequence of the nature of selection in our scenario. The alleles underlying individual trait value were of large effect and a low number. With a total of $10^{5}$ loci, to be realistic, each loci should represent 1000 $Kb$ in real data.

We have also shown introgression is beneficial for inferring causative loci from divergence ($F_{st}$) along the genome. 
This is generally because noise of selectively neutral alleles divergence can appear causative when genetic drift causes more 
differences between populations that have little gene flow between them.
It's important to know that in all scenarios, hitchhiking of selectively neutral alleles could also be 
mistaken for being causative as they often display the same amount of divergence.

%wouldn't it be cool if we could use this fact and simulate introgression, starting with real data that is noisy to extrapolate the areas we care about?
%this could be nonsense 

% at low M we should be able to predict something ---
% at high M we should be able to predict something ----

%Here, we suggest a range of migration rate (introgression) parameter values which would allow for the rapid (and in turn, parallel) adaptation 
%of marine stickleback introduced intro a freshwater environment.
%While this range is heavily effected by all other selections of parameter values. 

%(1) There's a threshold of introgression for both rapid adaptation and migration load. 
%(1.5) Maybe refer to the literature for migration load.  
\subsection*{thresholds}

We have found that too little migration leads to selection upon new mutations in all subpopulations and lakes alike. In contrast, at high migration rates we have seen that migration load limits the ability for species to locally adapt to the selective pressure of their environment. This leads us to consider a window of introgression which allows for the transportation of FAA's without migration load. 

\subsection*{connect results back to real data?}

%In a species that commonly is subjected to two general types of selective pressure 
%such as marine and freshwater stickleback this is suggestive of the benefit of hybridization in all subpopulations before
%reaching a point of pre or post zygotic 

%(3) discuss what signals researchers could possibly look out for or connect to real data

\paragraph{The adaptive filter?}
RAMBLING THOUGHTS HERE
Since larger effect alleles are more likely to establish,
be it by mutation or migration,
repeated colonization of new freshwater habitats will select for larger alleles,
be it single alleles or haplotypes bound together by an inversion.
However, these are more strongly selected against in the interstitial time.
Being recessive would help with this,
but would also make it more difficult to establish.

\jgg{TODO: effect of smaller than realistic population sizes?}

\jgg{TODO: Talk about the role recombination plays}
\jgg{Below may belong in the discussion / future work}
This also brings to light the role of recombination in a system of migration selection balance. 
On one hand, a lower recombination rate would allow for the freshwater haplotype to remain "intact" 
within the marine environment, however, without significant introgression the haplotype would quickly get selected against in marine populations. 
On the other hand a higher recombination rate would, in theory, allow freshwater adapted alleles more longevity 
as they would hitchhike with marine adapted haplotypes. 


\paragraph{Modeling assumptions}
Our simulations had much smaller population sizes than real populations.
How might this affect things?

\bibliographystyle{plainnat}
\bibliography{Citations}{}

%%%%%%%%%%%%%%%%%%%%%%%%%
\clearpage
\appendix
\setcounter{table}{0}
\renewcommand{\thetable}{S\arabic{table}}
\setcounter{figure}{0}
\renewcommand{\thefigure}{S\arabic{figure}}


\section*{Supplementary material}

\begin{figure}
	\begin{center}
  		\includegraphics{Final_Plots/True_Power.pdf}
  		\caption{ 
		Statistical Power and False Positives as a function of $F_{st}$ threshold. 
		Statistical Power is the likelihood that a SNP will be predicted to have an effect on phenotype when there is an effect to be detected (?).
		False Positives give us the ratio of SNPs that effect phenotype to total SNPs greater than the $F_{st}$ threshold.
            \textbf{
                TODO: fix this caption; x-axis label ($F_{ST}$).}
		}
  		\label{fig:Power_FP}
	\end{center}
\end{figure}

\begin{figure}
	\begin{center}
  		\includegraphics[width=1.0\linewidth]{Final_Plots/Haplo_small.pdf}
  		\caption{
			Phenotypes (\textbf{Top}) (effect size and mutation type) and haplotypes \textbf{(Botton)} as a function of spatial position of the individual 
			possessing the genome. The phenotype of each haplotype is, by definition, the sum of the 
			effect sizes of each mutation in the genome. 
		}
  		\label{fig:Haplo_Pheno}
	\end{center}
\end{figure}

\begin{figure}
	\begin{center}
        \includegraphics[width=1.0\linewidth]{plotlyPlots/PhenotypeThroughout5e-2.png}
        \includegraphics[width=1.0\linewidth]{plotlyPlots/PhenotypeThroughout5e-3.png}
  		\caption{
            \textbf{Mean phenotypes through time},
            at migration rates of
            \textbf{(top)} 10 and
            \textbf{(bottom)} 1 migrants per lake per generation.
		}
  		\label{fig:pheno_by_time1}
	\end{center}
\end{figure}

\begin{figure}
	\begin{center}
        \includegraphics[width=1.0\linewidth]{plotlyPlots/PhenotypeThroughout5e-4.png}
        \includegraphics[width=1.0\linewidth]{plotlyPlots/PhenotypeThroughout5e-5.png}
  		\caption{
            \textbf{Mean phenotypes through time},
            at migration rates of
            \textbf{(top)} 0.1 and
            \textbf{(bottom)} 0.01 migrants per lake per generation.
		}
  		\label{fig:pheno_by_time2}
	\end{center}
\end{figure}

\begin{figure}
	\begin{center}
        \includegraphics[width=1.0\linewidth]{plotlyPlots/StandingGeneticVariation.png}
  		\caption{
            \textbf{Standing genetic variance},
            computed as XXX write if we leave this in XXX.
		}
        \label{fig:SGV}
	\end{center}
\end{figure}




\end{document}
